---
title: "Class 11: Structural Bioinformatics pt2"
author: "Kavi Gonur (PID: A69046927)"
format: gfm
---

## Background

We saw last day that the PDB has 209,886 entries (Oct/Nov 2025). UniProtKB (i.e. protein sequence database) has 199,579,901 entries.

PDB has `r 209886 / 199579901 * 100`% coverage of the main sequence database.

```{r}
209886 / 199579901 * 100
```
Enter AlphaFold database (AFDB). <https://alphafold.ebi.ac.uk> that attempts to provide computed models for all sequences in UniProt.

"AlphaFold DB provides open access to over 200 million protein structure predictions to accelerate scientific research."


## AlphaFold

AlphaFold has 3 main outputs
- the predicted coordinates (PDB files)
- a local quality score called **pLDDT** (one for each amino-acid)
- a second quality score **PAE** Predicted Aligned Error (for each pair of amino acid)

We can run AlphaFold ourselves if we not happy with AFDB (i.e. no coverage or poor model).

```{r}
results_dir <- "HIVPR_dimer_23119.result/HIVPR_dimer_23119/"
pdb_files <- list.files(path=results_dir,
                        pattern="*.pdb",
                        full.names = TRUE)
basename(pdb_files)
```
```{r}
library(bio3d)
pdbs <- pdbaln(pdb_files, fit=TRUE, exefile="msa")
```
```{r}
rd <- rmsd(pdbs, fit=T)
range(rd)
```
```{r}
library(pheatmap)

colnames(rd) <- paste0("m",1:5)
rownames(rd) <- paste0("m",1:5)
pheatmap(rd)
```
```{r}
pdb <- read.pdb("1hsg")
plotb3(pdbs$b[1,], typ="l", lwd=2, sse=pdb)
points(pdbs$b[2,], typ="l", col="red")
points(pdbs$b[3,], typ="l", col="blue")
points(pdbs$b[4,], typ="l", col="darkgreen")
points(pdbs$b[5,], typ="l", col="orange")
abline(v=100, col="gray")
```
```{r}
core <- core.find(pdbs)
core.inds <- print(core, vol=0.5)
xyz <- pdbfit(pdbs, core.inds, outpath="corefit_structures")
rf <- rmsf(xyz)

plotb3(rf, sse=pdb)
abline(v=100, col="gray", ylab="RMSF")
```

