<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.32">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Kavi (PID: A69046927)">

<title>Class 9</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="class09_files/libs/clipboard/clipboard.min.js"></script>
<script src="class09_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="class09_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="class09_files/libs/quarto-html/popper.min.js"></script>
<script src="class09_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="class09_files/libs/quarto-html/anchor.min.js"></script>
<link href="class09_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="class09_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="class09_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="class09_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="class09_files/libs/bootstrap/bootstrap-bb462d781dde1847d9e3ccf7736099dd.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">


</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#visualizing-structure-data" id="toc-visualizing-structure-data" class="nav-link active" data-scroll-target="#visualizing-structure-data">Visualizing Structure Data</a>
  <ul class="collapse">
  <li><a href="#align-and-superpose-structures" id="toc-align-and-superpose-structures" class="nav-link" data-scroll-target="#align-and-superpose-structures">Align and superpose structures</a></li>
  <li><a href="#principal-component-analysis" id="toc-principal-component-analysis" class="nav-link" data-scroll-target="#principal-component-analysis">Principal component analysis</a></li>
  </ul></li>
  <li><a href="#prediction-of-functional-motions" id="toc-prediction-of-functional-motions" class="nav-link" data-scroll-target="#prediction-of-functional-motions">Prediction of functional motions</a></li>
  <li><a href="#play-with-3d-viewing-in-r" id="toc-play-with-3d-viewing-in-r" class="nav-link" data-scroll-target="#play-with-3d-viewing-in-r">Play with 3D viewing in R</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Class 9</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Kavi (PID: A69046927) </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"PDB_Data_Dis.csv"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">","</span>,<span class="st">""</span>, stats<span class="sc">$</span>X.ray))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 176378  10284   9007   3077    174     11</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">as.numeric</span>(<span class="fu">sub</span>(<span class="st">","</span>,<span class="st">""</span>, stats<span class="sc">$</span>Total))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 210203  13734  15256   4852    223     22</code></pre>
</div>
</div>
<p>This is annoying. Let’s see if we can make it easier with a different package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readr' was built under R version 4.5.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>stats <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">"PDB_Data_Dis.csv"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 6 Columns: 9
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (1): Molecular Type
dbl (4): Integrative, Multiple methods, Neutron, Other
num (4): X-ray, EM, NMR, Total

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>stats</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 9
  `Molecular Type`    `X-ray`    EM   NMR Integrative `Multiple methods` Neutron
  &lt;chr&gt;                 &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;       &lt;dbl&gt;              &lt;dbl&gt;   &lt;dbl&gt;
1 Protein (only)       176378 20438 12709         342                221      83
2 Protein/Oligosacch…   10284  3396    34           8                 11       1
3 Protein/NA             9007  5931   287          24                  7       0
4 Nucleic acid (only)    3077   200  1554           2                 15       3
5 Other                   174    13    33           3                  0       0
6 Oligosaccharide (o…      11     0     6           0                  1       0
# ℹ 2 more variables: Other &lt;dbl&gt;, Total &lt;dbl&gt;</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>n.total <span class="ot">&lt;-</span> <span class="fu">sum</span>(stats<span class="sc">$</span>Total)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>n.xray <span class="ot">&lt;-</span> <span class="fu">sum</span>(stats<span class="sc">$</span><span class="st">`</span><span class="at">X-ray</span><span class="st">`</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>n.em <span class="ot">&lt;-</span> <span class="fu">sum</span>(stats<span class="sc">$</span>EM)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(n.xray<span class="sc">/</span>n.total <span class="sc">*</span> <span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 81.43</code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">round</span>(n.em<span class="sc">/</span>n.total <span class="sc">*</span> <span class="dv">100</span>,<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 12.27</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(stats, <span class="fu">aes</span>(<span class="fu">reorder</span>(<span class="st">`</span><span class="at">Molecular Type</span><span class="st">`</span>,<span class="sc">-</span>Total),Total)) <span class="sc">+</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Q2: What proportion of structures in the PDB are protein?</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>stats<span class="sc">$</span><span class="st">`</span><span class="at">Molecular Type</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">trimws</span>(stats<span class="sc">$</span><span class="st">`</span><span class="at">Molecular Type</span><span class="st">`</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>n.total <span class="ot">&lt;-</span> <span class="fu">sum</span>(stats<span class="sc">$</span>Total)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>n.protein <span class="ot">&lt;-</span> <span class="fu">sum</span>(stats<span class="sc">$</span>Total[<span class="fu">grepl</span>(<span class="st">"Protein"</span>, stats<span class="sc">$</span><span class="st">`</span><span class="at">Molecular Type</span><span class="st">`</span>, <span class="at">ignore.case =</span> <span class="cn">TRUE</span>)])</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>prot.por <span class="ot">&lt;-</span> <span class="fu">round</span>(n.protein <span class="sc">/</span> n.total <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>97.91% of structures in the PDB are protein.</p>
<blockquote class="blockquote">
<p>Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?</p>
</blockquote>
<p>363 structures</p>
<section id="visualizing-structure-data" class="level2">
<h2 class="anchored" data-anchor-id="visualizing-structure-data">Visualizing Structure Data</h2>
<p>The Mol* viewer is embedded in many bioinformatics websites. The homepage is https://molstar.org</p>
<p>I can insert any figure or image file unsing markdown format</p>
<p><img src="1HSG.png" class="img-fluid" alt="The HIV-Pr dimer with bound inhibitor"> <img src="1HSG-SF.png" class="img-fluid" alt="The HIV-Pr dimer with bound inhibitor, water molecule, and active regions in spacefill"> <img src="1HSG-FULL.png" class="img-fluid" alt="Another view"> ## Bio3D package for structural bioinformatics</p>
<p>We can use the bio3d package to read and analyze biomoecular data in R:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bio3d)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>hiv <span class="ot">&lt;-</span> <span class="fu">read.pdb</span>(<span class="st">"1hsg"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Note: Accessing on-line PDB file</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>hiv</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 Call:  read.pdb(file = "1hsg")

   Total Models#: 1
     Total Atoms#: 1686,  XYZs#: 5058  Chains#: 2  (values: A B)

     Protein Atoms#: 1514  (residues/Calpha atoms#: 198)
     Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)

     Non-protein/nucleic Atoms#: 172  (residues: 128)
     Non-protein/nucleic resid values: [ HOH (127), MK1 (1) ]

   Protein sequence:
      PQITLWQRPLVTIKIGGQLKEALLDTGADDTVLEEMSLPGRWKPKMIGGIGGFIKVRQYD
      QILIEICGHKAIGTVLVGPTPVNIIGRNLLTQIGCTLNFPQITLWQRPLVTIKIGGQLKE
      ALLDTGADDTVLEEMSLPGRWKPKMIGGIGGFIKVRQYDQILIEICGHKAIGTVLVGPTP
      VNIIGRNLLTQIGCTLNF

+ attr: atom, xyz, seqres, helix, sheet,
        calpha, remark, call</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(hiv<span class="sc">$</span>atom)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  type eleno elety  alt resid chain resno insert      x      y     z o     b
1 ATOM     1     N &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 29.361 39.686 5.862 1 38.10
2 ATOM     2    CA &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 30.307 38.663 5.319 1 40.62
3 ATOM     3     C &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 29.760 38.071 4.022 1 42.64
4 ATOM     4     O &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 28.600 38.302 3.676 1 43.40
5 ATOM     5    CB &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 30.508 37.541 6.342 1 37.87
6 ATOM     6    CG &lt;NA&gt;   PRO     A     1   &lt;NA&gt; 29.296 37.591 7.162 1 38.40
  segid elesy charge
1  &lt;NA&gt;     N   &lt;NA&gt;
2  &lt;NA&gt;     C   &lt;NA&gt;
3  &lt;NA&gt;     C   &lt;NA&gt;
4  &lt;NA&gt;     O   &lt;NA&gt;
5  &lt;NA&gt;     C   &lt;NA&gt;
6  &lt;NA&gt;     C   &lt;NA&gt;</code></pre>
</div>
</div>
<p>Let’s get the sequence</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pdbseq</span>(hiv)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20 
"P" "Q" "I" "T" "L" "W" "Q" "R" "P" "L" "V" "T" "I" "K" "I" "G" "G" "Q" "L" "K" 
 21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40 
"E" "A" "L" "L" "D" "T" "G" "A" "D" "D" "T" "V" "L" "E" "E" "M" "S" "L" "P" "G" 
 41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60 
"R" "W" "K" "P" "K" "M" "I" "G" "G" "I" "G" "G" "F" "I" "K" "V" "R" "Q" "Y" "D" 
 61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80 
"Q" "I" "L" "I" "E" "I" "C" "G" "H" "K" "A" "I" "G" "T" "V" "L" "V" "G" "P" "T" 
 81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99   1 
"P" "V" "N" "I" "I" "G" "R" "N" "L" "L" "T" "Q" "I" "G" "C" "T" "L" "N" "F" "P" 
  2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21 
"Q" "I" "T" "L" "W" "Q" "R" "P" "L" "V" "T" "I" "K" "I" "G" "G" "Q" "L" "K" "E" 
 22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41 
"A" "L" "L" "D" "T" "G" "A" "D" "D" "T" "V" "L" "E" "E" "M" "S" "L" "P" "G" "R" 
 42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61 
"W" "K" "P" "K" "M" "I" "G" "G" "I" "G" "G" "F" "I" "K" "V" "R" "Q" "Y" "D" "Q" 
 62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81 
"I" "L" "I" "E" "I" "C" "G" "H" "K" "A" "I" "G" "T" "V" "L" "V" "G" "P" "T" "P" 
 82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 
"V" "N" "I" "I" "G" "R" "N" "L" "L" "T" "Q" "I" "G" "C" "T" "L" "N" "F" </code></pre>
</div>
</div>
<p>Let’s trim to chain A and get just its sequence:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>chainA <span class="ot">&lt;-</span> <span class="fu">trim.pdb</span>(hiv,<span class="at">chain=</span><span class="st">"A"</span>)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>chainA.seq <span class="ot">&lt;-</span> <span class="fu">pdbseq</span>(chainA)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s blast</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>blast <span class="ot">&lt;-</span> <span class="fu">blast.pdb</span>(chainA.seq)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Searching ... please wait (updates every 5 seconds) RID = GH5ZUHK8016 
 .
 Reporting 249 hits</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(blast<span class="sc">$</span>hit.tbl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>        queryid subjectids identity alignmentlength mismatches gapopens q.start
1 Query_3031849     1W5V_A   100.00              99          0        0       1
2 Query_3031849     2FDE_A   100.00              99          0        0       1
3 Query_3031849     1AJV_A   100.00              99          0        0       1
4 Query_3031849     2R38_A    98.99              99          1        0       1
5 Query_3031849     2R3T_A    98.99              99          1        0       1
6 Query_3031849     1HXB_A    98.99              99          1        0       1
  q.end s.start s.end   evalue bitscore positives mlog.evalue pdb.id    acc
1    99      12   110 1.38e-67      199       100    153.9511 1W5V_A 1W5V_A
2    99       2   100 1.70e-67      198       100    153.7426 2FDE_A 2FDE_A
3    99       1    99 1.99e-67      198       100    153.5851 1AJV_A 1AJV_A
4    99       1    99 2.50e-67      198       100    153.3569 2R38_A 2R38_A
5    99       1    99 2.50e-67      198       100    153.3569 2R3T_A 2R3T_A
6    99       1    99 2.50e-67      198       100    153.3569 1HXB_A 1HXB_A</code></pre>
</div>
</div>
<p>Plot a quick overview of blast results</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>hits <span class="ot">&lt;-</span> <span class="fu">plot</span>(blast)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  * Possible cutoff values:    135 110 69 -2 
            Yielding Nhits:    186 238 244 249 

  * Chosen cutoff value of:    69 
            Yielding Nhits:    244 </code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>hits<span class="sc">$</span>pdb.id</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  [1] "1W5V_A" "2FDE_A" "1AJV_A" "2R38_A" "2R3T_A" "1HXB_A" "1BV9_A" "1AAQ_A"
  [9] "1AXA_A" "1HVS_A" "1ZP8_A" "2QHC_A" "1A8G_A" "2O4L_A" "5COK_A" "1TCX_A"
 [17] "2Z54_A" "1D4S_A" "1BV7_A" "1BWA_A" "1A9M_A" "2FLE_A" "1ODY_A" "1GNN_A"
 [25] "1GNM_A" "5YRS_B" "1HEF_E" "1ODX_A" "4QGI_A" "1BVE_A" "2AZ8_A" "1A30_A"
 [33] "6DH6_A" "6DH0_A" "2I4D_A" "6OOS_A" "1RL8_A" "5YRS_A" "1ZSF_A" "2Q64_A"
 [41] "6DH3_A" "2NPH_A" "2Q63_A" "1LZQ_A" "1FB7_A" "1G6L_A" "1HIV_A" "6OOU_A"
 [49] "1HVC_A" "2I4V_A" "2AZ9_A" "6OOT_A" "2P3B_B" "5KAO_A" "2WL0_A" "6OPT_A"
 [57] "1IZI_A" "1MRX_A" "2PYM_A" "2PYN_A" "1DMP_A" "4K4P_A" "1LV1_A" "1AID_A"
 [65] "1LV1_A" "1ZBG_A" "3TKG_A" "1HVC_A" "5YOK_A" "1G6L_A" "1FGC_C" "3K4V_A"
 [73] "3KT5_A" "3KT5_A" "4QLH_A" "4QLH_A" "2F3K_A" "4Q5M_A" "2AOC_A" "3B80_A"
 [81] "3VF5_A" "2AVQ_A" "1DW6_C" "1KZK_A" "2HS1_A" "1K6C_A" "1MTB_A" "4Q1X_A"
 [89] "4Q1W_A" "4Q5M_A" "3D1X_A" "2AVM_A" "3PWM_A" "3KT2_A" "3KT2_A" "1SDV_A"
 [97] "3JVW_A" "3OY4_A" "1A94_A" "2HS2_A" "4EJ8_A" "2FGU_A" "2AVV_A" "3JW2_A"
[105] "3BVA_A" "1FFF_C" "3S43_B" "2NXD_A" "1FG6_C" "1EBK_C" "4Q1Y_A" "3EL4_A"
[113] "1F7A_A" "1K2B_A" "2FGV_A" "1Z8C_A" "2G69_A" "3EL9_A" "3OXV_A" "1BDR_A"
[121] "3N3I_A" "3N3I_A" "3OXW_A" "3S43_A" "3EM3_A" "3CYW_A" "5KQX_A" "2B60_A"
[129] "7DOZ_A" "1K2C_A" "1MT7_A" "3EM4_A" "4QJ9_A" "1BDL_A" "3LZS_A" "5T84_A"
[137] "4DQB_A" "7DOZ_A" "4QJ2_A" "3LZV_A" "1SGU_A" "2FXE_A" "1BDQ_A" "3U71_A"
[145] "2R5P_A" "4OBD_A" "7MAS_A" "3IXO_A" "3D3T_A" "5YOJ_A" "3LZU_A" "4NJS_A"
[153] "3EKP_A" "1B6J_A" "3EKQ_A" "2RKF_A" "1C6X_A" "7MAR_A" "4DQF_A" "1RPI_A"
[161] "3OU1_B" "3PJ6_A" "2P3A_A" "6OGQ_A" "3OQ7_A" "5KR1_A" "3OQD_A" "4RVI_A"
[169] "3OQA_A" "1B6K_A" "3OUD_B" "6MK9_A" "3SO9_A" "1Q9P_A" "6I45_A" "7SEP_A"
[177] "4NJT_A" "3BXR_A" "4YOA_A" "4DQC_A" "2FDD_A" "2RKG_A" "4DQH_A" "2P3C_A"
[185] "4EP2_A" "4EP2_A" "4EQ0_A" "4NPT_A" "6OPU_A" "4NPU_A" "3U7S_A" "3HAW_A"
[193] "2AZB_A" "3TTP_A" "3HBO_A" "3GGU_A" "7N6T_A" "6OPV_A" "4EQ0_A" "6OPX_A"
[201] "2O4N_A" "5T2E_A" "3UCB_A" "3KA2_A" "3FSM_A" "6OPW_A" "2AZC_A" "3FSM_A"
[209] "3HLO_A" "2P3D_A" "3T3C_A" "7MYP_A" "6O54_X" "6OPY_A" "4Z4X_A" "6OPZ_A"
[217] "2JE4_A" "1DAZ_C" "7MAP_A" "7MAQ_A" "1K1U_A" "2B7Z_A" "3MWS_A" "1K1T_A"
[225] "8DCH_A" "3I2L_A" "6P9A_A" "2FXD_A" "2J9J_A" "3DCK_A" "2J9J_B" "3NXE_A"
[233] "2O40_A" "2O40_A" "3NXE_A" "3KA2_A" "3HLO_A" "5B18_A" "1SIP_A" "2SAM_A"
[241] "1AZ5_A" "1SIV_A" "1HII_A" "1IVP_A"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">get.pdb</span>(hits<span class="sc">$</span>pdb.id[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>], <span class="at">path =</span> <span class="st">"pdbs"</span>, <span class="at">split =</span> <span class="cn">TRUE</span>, <span class="at">gzip =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1W5V.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2FDE.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1AJV.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2R38.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2R3T.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1HXB.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1BV9.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1AAQ.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1AXA.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1HVS.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1ZP8.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2QHC.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1A8G.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2O4L.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/5COK.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1TCX.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2Z54.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1D4S.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1BV7.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1BWA.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1A9M.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2FLE.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1ODY.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1GNN.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1GNM.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/5YRS.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1HEF.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1ODX.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4QGI.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1BVE.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2AZ8.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1A30.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6DH6.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6DH0.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2I4D.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6OOS.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1RL8.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1ZSF.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2Q64.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6DH3.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2NPH.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2Q63.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1LZQ.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1FB7.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1G6L.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1HIV.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6OOU.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1HVC.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2I4V.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2AZ9.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6OOT.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2P3B.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/5KAO.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2WL0.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/6OPT.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1IZI.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1MRX.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2PYM.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2PYN.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1DMP.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4K4P.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1LV1.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1AID.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1ZBG.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3TKG.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/5YOK.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1FGC.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3K4V.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3KT5.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4QLH.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2F3K.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4Q5M.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2AOC.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3B80.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3VF5.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2AVQ.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1DW6.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1KZK.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2HS1.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1K6C.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1MTB.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4Q1X.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/4Q1W.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3D1X.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2AVM.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3PWM.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3KT2.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1SDV.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3JVW.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/3OY4.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/1A94.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in get.pdb(hits$pdb.id[1:100], path = "pdbs", split = TRUE, gzip =
TRUE): pdbs/2HS2.pdb exists. Skipping download</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   1%
  |                                                                            
  |==                                                                    |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |==============                                                        |  21%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |==================                                                    |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |===================                                                   |  28%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  29%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |========================                                              |  35%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |===========================                                           |  38%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |==============================                                        |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  45%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |=================================                                     |  48%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  55%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |========================================                              |  58%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |===========================================                           |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |================================================                      |  68%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |====================================================                  |  75%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  79%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |====================================================================  |  98%
  |                                                                            
  |===================================================================== |  99%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
</div>
<section id="align-and-superpose-structures" class="level3">
<h3 class="anchored" data-anchor-id="align-and-superpose-structures">Align and superpose structures</h3>
<p>Next we will use the pdbaln() function to align and also optionally fit (i.e.&nbsp;superpose) the identified PDB structures.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb128"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb128-1"><a href="#cb128-1" aria-hidden="true" tabindex="-1"></a>pdbs <span class="ot">&lt;-</span> <span class="fu">pdbaln</span>(files, <span class="at">fit =</span> <span class="cn">TRUE</span>, <span class="at">exefile=</span><span class="st">"msa"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Reading PDB files:
pdbs/split_chain/1W5V_A.pdb
pdbs/split_chain/2FDE_A.pdb
pdbs/split_chain/1AJV_A.pdb
pdbs/split_chain/2R38_A.pdb
pdbs/split_chain/2R3T_A.pdb
pdbs/split_chain/1HXB_A.pdb
pdbs/split_chain/1BV9_A.pdb
pdbs/split_chain/1AAQ_A.pdb
pdbs/split_chain/1AXA_A.pdb
pdbs/split_chain/1HVS_A.pdb
pdbs/split_chain/1ZP8_A.pdb
pdbs/split_chain/2QHC_A.pdb
pdbs/split_chain/1A8G_A.pdb
pdbs/split_chain/2O4L_A.pdb
pdbs/split_chain/5COK_A.pdb
pdbs/split_chain/1TCX_A.pdb
pdbs/split_chain/2Z54_A.pdb
pdbs/split_chain/1D4S_A.pdb
pdbs/split_chain/1BV7_A.pdb
pdbs/split_chain/1BWA_A.pdb
pdbs/split_chain/1A9M_A.pdb
pdbs/split_chain/2FLE_A.pdb
pdbs/split_chain/1ODY_A.pdb
pdbs/split_chain/1GNN_A.pdb
pdbs/split_chain/1GNM_A.pdb
pdbs/split_chain/5YRS_B.pdb
pdbs/split_chain/5YRS_A.pdb
pdbs/split_chain/1HEF_E.pdb
pdbs/split_chain/1ODX_A.pdb
pdbs/split_chain/4QGI_A.pdb
pdbs/split_chain/1BVE_A.pdb
pdbs/split_chain/2AZ8_A.pdb
pdbs/split_chain/1A30_A.pdb
pdbs/split_chain/6DH6_A.pdb
pdbs/split_chain/6DH0_A.pdb
pdbs/split_chain/2I4D_A.pdb
pdbs/split_chain/6OOS_A.pdb
pdbs/split_chain/1RL8_A.pdb
pdbs/split_chain/1ZSF_A.pdb
pdbs/split_chain/2Q64_A.pdb
pdbs/split_chain/6DH3_A.pdb
pdbs/split_chain/2NPH_A.pdb
pdbs/split_chain/2Q63_A.pdb
pdbs/split_chain/1LZQ_A.pdb
pdbs/split_chain/1FB7_A.pdb
pdbs/split_chain/1G6L_A.pdb
pdbs/split_chain/1HIV_A.pdb
pdbs/split_chain/6OOU_A.pdb
pdbs/split_chain/1HVC_A.pdb
pdbs/split_chain/2I4V_A.pdb
pdbs/split_chain/2AZ9_A.pdb
pdbs/split_chain/6OOT_A.pdb
pdbs/split_chain/2P3B_B.pdb
pdbs/split_chain/5KAO_A.pdb
pdbs/split_chain/2WL0_A.pdb
pdbs/split_chain/6OPT_A.pdb
pdbs/split_chain/1IZI_A.pdb
pdbs/split_chain/1MRX_A.pdb
pdbs/split_chain/2PYM_A.pdb
pdbs/split_chain/2PYN_A.pdb
pdbs/split_chain/1DMP_A.pdb
pdbs/split_chain/4K4P_A.pdb
pdbs/split_chain/1LV1_A.pdb
pdbs/split_chain/1AID_A.pdb
pdbs/split_chain/1ZBG_A.pdb
pdbs/split_chain/3TKG_A.pdb
pdbs/split_chain/5YOK_A.pdb
pdbs/split_chain/1FGC_C.pdb
pdbs/split_chain/3K4V_A.pdb
pdbs/split_chain/3KT5_A.pdb
pdbs/split_chain/4QLH_A.pdb
pdbs/split_chain/2F3K_A.pdb
pdbs/split_chain/4Q5M_A.pdb
pdbs/split_chain/2AOC_A.pdb
pdbs/split_chain/3B80_A.pdb
pdbs/split_chain/3VF5_A.pdb
pdbs/split_chain/2AVQ_A.pdb
pdbs/split_chain/1DW6_C.pdb
pdbs/split_chain/1KZK_A.pdb
pdbs/split_chain/2HS1_A.pdb
pdbs/split_chain/1K6C_A.pdb
pdbs/split_chain/1MTB_A.pdb
pdbs/split_chain/4Q1X_A.pdb
pdbs/split_chain/4Q1W_A.pdb
pdbs/split_chain/3D1X_A.pdb
pdbs/split_chain/2AVM_A.pdb
pdbs/split_chain/3PWM_A.pdb
pdbs/split_chain/3KT2_A.pdb
pdbs/split_chain/1SDV_A.pdb
pdbs/split_chain/3JVW_A.pdb
pdbs/split_chain/3OY4_A.pdb
pdbs/split_chain/1A94_A.pdb
pdbs/split_chain/2HS2_A.pdb
.....   PDB has ALT records, taking A only, rm.alt=TRUE
...   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.............   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
....   PDB has ALT records, taking A only, rm.alt=TRUE
...   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
...   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.....   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking 1 only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
.   PDB has ALT records, taking A only, rm.alt=TRUE
..   PDB has ALT records, taking A only, rm.alt=TRUE
.

Extracting sequences

pdb/seq: 1   name: pdbs/split_chain/1W5V_A.pdb 
pdb/seq: 2   name: pdbs/split_chain/2FDE_A.pdb 
pdb/seq: 3   name: pdbs/split_chain/1AJV_A.pdb 
pdb/seq: 4   name: pdbs/split_chain/2R38_A.pdb 
pdb/seq: 5   name: pdbs/split_chain/2R3T_A.pdb 
pdb/seq: 6   name: pdbs/split_chain/1HXB_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 7   name: pdbs/split_chain/1BV9_A.pdb 
pdb/seq: 8   name: pdbs/split_chain/1AAQ_A.pdb 
pdb/seq: 9   name: pdbs/split_chain/1AXA_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 10   name: pdbs/split_chain/1HVS_A.pdb 
pdb/seq: 11   name: pdbs/split_chain/1ZP8_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 12   name: pdbs/split_chain/2QHC_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 13   name: pdbs/split_chain/1A8G_A.pdb 
pdb/seq: 14   name: pdbs/split_chain/2O4L_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 15   name: pdbs/split_chain/5COK_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 16   name: pdbs/split_chain/1TCX_A.pdb 
pdb/seq: 17   name: pdbs/split_chain/2Z54_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 18   name: pdbs/split_chain/1D4S_A.pdb 
pdb/seq: 19   name: pdbs/split_chain/1BV7_A.pdb 
pdb/seq: 20   name: pdbs/split_chain/1BWA_A.pdb 
pdb/seq: 21   name: pdbs/split_chain/1A9M_A.pdb 
pdb/seq: 22   name: pdbs/split_chain/2FLE_A.pdb 
pdb/seq: 23   name: pdbs/split_chain/1ODY_A.pdb 
pdb/seq: 24   name: pdbs/split_chain/1GNN_A.pdb 
pdb/seq: 25   name: pdbs/split_chain/1GNM_A.pdb 
pdb/seq: 26   name: pdbs/split_chain/5YRS_B.pdb 
pdb/seq: 27   name: pdbs/split_chain/5YRS_A.pdb 
pdb/seq: 28   name: pdbs/split_chain/1HEF_E.pdb 
pdb/seq: 29   name: pdbs/split_chain/1ODX_A.pdb 
pdb/seq: 30   name: pdbs/split_chain/4QGI_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 31   name: pdbs/split_chain/1BVE_A.pdb 
pdb/seq: 32   name: pdbs/split_chain/2AZ8_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 33   name: pdbs/split_chain/1A30_A.pdb 
pdb/seq: 34   name: pdbs/split_chain/6DH6_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 35   name: pdbs/split_chain/6DH0_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 36   name: pdbs/split_chain/2I4D_A.pdb 
pdb/seq: 37   name: pdbs/split_chain/6OOS_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 38   name: pdbs/split_chain/1RL8_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 39   name: pdbs/split_chain/1ZSF_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 40   name: pdbs/split_chain/2Q64_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 41   name: pdbs/split_chain/6DH3_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 42   name: pdbs/split_chain/2NPH_A.pdb 
pdb/seq: 43   name: pdbs/split_chain/2Q63_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 44   name: pdbs/split_chain/1LZQ_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 45   name: pdbs/split_chain/1FB7_A.pdb 
pdb/seq: 46   name: pdbs/split_chain/1G6L_A.pdb 
pdb/seq: 47   name: pdbs/split_chain/1HIV_A.pdb 
pdb/seq: 48   name: pdbs/split_chain/6OOU_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 49   name: pdbs/split_chain/1HVC_A.pdb 
pdb/seq: 50   name: pdbs/split_chain/2I4V_A.pdb 
pdb/seq: 51   name: pdbs/split_chain/2AZ9_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 52   name: pdbs/split_chain/6OOT_A.pdb 
pdb/seq: 53   name: pdbs/split_chain/2P3B_B.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 54   name: pdbs/split_chain/5KAO_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 55   name: pdbs/split_chain/2WL0_A.pdb 
pdb/seq: 56   name: pdbs/split_chain/6OPT_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 57   name: pdbs/split_chain/1IZI_A.pdb 
pdb/seq: 58   name: pdbs/split_chain/1MRX_A.pdb 
pdb/seq: 59   name: pdbs/split_chain/2PYM_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 60   name: pdbs/split_chain/2PYN_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 61   name: pdbs/split_chain/1DMP_A.pdb 
pdb/seq: 62   name: pdbs/split_chain/4K4P_A.pdb 
pdb/seq: 63   name: pdbs/split_chain/1LV1_A.pdb 
pdb/seq: 64   name: pdbs/split_chain/1AID_A.pdb 
pdb/seq: 65   name: pdbs/split_chain/1ZBG_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 66   name: pdbs/split_chain/3TKG_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 67   name: pdbs/split_chain/5YOK_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 68   name: pdbs/split_chain/1FGC_C.pdb 
pdb/seq: 69   name: pdbs/split_chain/3K4V_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 70   name: pdbs/split_chain/3KT5_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 71   name: pdbs/split_chain/4QLH_A.pdb 
pdb/seq: 72   name: pdbs/split_chain/2F3K_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 73   name: pdbs/split_chain/4Q5M_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 74   name: pdbs/split_chain/2AOC_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 75   name: pdbs/split_chain/3B80_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 76   name: pdbs/split_chain/3VF5_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 77   name: pdbs/split_chain/2AVQ_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 78   name: pdbs/split_chain/1DW6_C.pdb 
pdb/seq: 79   name: pdbs/split_chain/1KZK_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 80   name: pdbs/split_chain/2HS1_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 81   name: pdbs/split_chain/1K6C_A.pdb 
   PDB has ALT records, taking 1 only, rm.alt=TRUE
pdb/seq: 82   name: pdbs/split_chain/1MTB_A.pdb 
pdb/seq: 83   name: pdbs/split_chain/4Q1X_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 84   name: pdbs/split_chain/4Q1W_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 85   name: pdbs/split_chain/3D1X_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 86   name: pdbs/split_chain/2AVM_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 87   name: pdbs/split_chain/3PWM_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 88   name: pdbs/split_chain/3KT2_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 89   name: pdbs/split_chain/1SDV_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 90   name: pdbs/split_chain/3JVW_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 91   name: pdbs/split_chain/3OY4_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE
pdb/seq: 92   name: pdbs/split_chain/1A94_A.pdb 
pdb/seq: 93   name: pdbs/split_chain/2HS2_A.pdb 
   PDB has ALT records, taking A only, rm.alt=TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>ids <span class="ot">&lt;-</span> <span class="fu">basename.pdb</span>(pdbs<span class="sc">$</span>id)</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pdbs, <span class="at">labels=</span>ids)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The function <code>pdb.annotate()</code> provides a convenient way of annotating the PDB files we have collected. Below we use the function to annotate each structure to its source species. This will come in handy when annotating plots later on:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb131"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb131-1"><a href="#cb131-1" aria-hidden="true" tabindex="-1"></a>anno <span class="ot">&lt;-</span> <span class="fu">pdb.annotate</span>(ids)</span>
<span id="cb131-2"><a href="#cb131-2" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(anno<span class="sc">$</span>source)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Human immunodeficiency virus"                                          
 [2] "Human immunodeficiency virus 1"                                        
 [3] "Human immunodeficiency virus type 1 (CLONE 12)"                        
 [4] "Human immunodeficiency virus type 1 (BH5 ISOLATE)"                     
 [5] "HIV-1 M:B_HXB2R"                                                       
 [6] "Human immunodeficiency virus type 1 lw12.3 isolate"                    
 [7] "Human immunodeficiency virus type 1 (Z2/CDC-Z34 ISOLATE)"              
 [8] "Human immunodeficiency virus type 1 (BRU ISOLATE)"                     
 [9] "Human immunodeficiency virus type 1 BH10"                              
[10] "Human immunodeficiency virus 1/synthetic construct"                    
[11] "NA/Human immunodeficiency virus type 1 group M subtype B (isolate BH5)"
[12] "HIV-1 M:B_ARV2/SF2"                                                    </code></pre>
</div>
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a>anno</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>       structureId chainId macromoleculeType chainLength experimentalTechnique
1W5V_A        1W5V       A           Protein         110                 X-ray
2FDE_A        2FDE       A           Protein         100                 X-ray
1AJV_A        1AJV       A           Protein          99                 X-ray
2R38_A        2R38       A           Protein          99                 X-ray
2R3T_A        2R3T       A           Protein          99                 X-ray
1HXB_A        1HXB       A           Protein          99                 X-ray
1BV9_A        1BV9       A           Protein          99                 X-ray
1AAQ_A        1AAQ       A           Protein          99                 X-ray
1AXA_A        1AXA       A           Protein          99                 X-ray
1HVS_A        1HVS       A           Protein          99                 X-ray
1ZP8_A        1ZP8       A           Protein          99                 X-ray
2QHC_A        2QHC       A           Protein          99                 X-ray
1A8G_A        1A8G       A           Protein          99                 X-ray
2O4L_A        2O4L       A           Protein          99                 X-ray
5COK_A        5COK       A           Protein          99                 X-ray
1TCX_A        1TCX       A           Protein          99                 X-ray
2Z54_A        2Z54       A           Protein          99                 X-ray
1D4S_A        1D4S       A           Protein          99                 X-ray
1BV7_A        1BV7       A           Protein          99                 X-ray
1BWA_A        1BWA       A           Protein          99                 X-ray
1A9M_A        1A9M       A           Protein          99                 X-ray
2FLE_A        2FLE       A           Protein          99                 X-ray
1ODY_A        1ODY       A           Protein          99                 X-ray
1GNN_A        1GNN       A           Protein          99                 X-ray
1GNM_A        1GNM       A           Protein          99                 X-ray
5YRS_B        5YRS       B           Protein         104                 X-ray
5YRS_A        5YRS       A           Protein         104                 X-ray
1HEF_E        1HEF       E           Protein          99                 X-ray
1ODX_A        1ODX       A           Protein          99                 X-ray
4QGI_A        4QGI       A           Protein          99                 X-ray
1BVE_A        1BVE       A           Protein          99                   NMR
2AZ8_A        2AZ8       A           Protein          99                 X-ray
1A30_A        1A30       A           Protein          99                 X-ray
6DH6_A        6DH6       A           Protein          99                 X-ray
6DH0_A        6DH0       A           Protein          99                 X-ray
2I4D_A        2I4D       A           Protein          99                 X-ray
6OOS_A        6OOS       A           Protein          99                 X-ray
1RL8_A        1RL8       A           Protein          99                 X-ray
1ZSF_A        1ZSF       A           Protein          99                 X-ray
2Q64_A        2Q64       A           Protein          99                 X-ray
6DH3_A        6DH3       A           Protein          99                 X-ray
2NPH_A        2NPH       A           Protein          99                 X-ray
2Q63_A        2Q63       A           Protein          99                 X-ray
1LZQ_A        1LZQ       A           Protein          99                 X-ray
1FB7_A        1FB7       A           Protein          99                 X-ray
1G6L_A        1G6L       A           Protein         203                 X-ray
1HIV_A        1HIV       A           Protein          99                 X-ray
6OOU_A        6OOU       A           Protein          99                 X-ray
1HVC_A        1HVC       A           Protein         203                 X-ray
2I4V_A        2I4V       A           Protein          99                 X-ray
2AZ9_A        2AZ9       A           Protein          99                 X-ray
6OOT_A        6OOT       A           Protein          99                 X-ray
2P3B_B        2P3B       B           Protein          99                 X-ray
5KAO_A        5KAO       A           Protein          99                 X-ray
2WL0_A        2WL0       A           Protein          99                 X-ray
6OPT_A        6OPT       A           Protein          99                 X-ray
1IZI_A        1IZI       A           Protein          99                 X-ray
1MRX_A        1MRX       A           Protein          99                 X-ray
2PYM_A        2PYM       A           Protein          99                 X-ray
2PYN_A        2PYN       A           Protein          99                 X-ray
1DMP_A        1DMP       A           Protein          99                 X-ray
4K4P_A        4K4P       A           Protein          99                 X-ray
1LV1_A        1LV1       A           Protein         203                 X-ray
1AID_A        1AID       A           Protein          99                 X-ray
1ZBG_A        1ZBG       A           Protein          99                 X-ray
3TKG_A        3TKG       A           Protein         103                 X-ray
5YOK_A        5YOK       A           Protein         100                 X-ray
1FGC_C        1FGC       C           Protein          99                 X-ray
3K4V_A        3K4V       A           Protein         100                 X-ray
3KT5_A        3KT5       A           Protein         203                 X-ray
4QLH_A        4QLH       A           Protein         203                 X-ray
2F3K_A        2F3K       A           Protein          99                 X-ray
4Q5M_A        4Q5M       A           Protein         203                 X-ray
2AOC_A        2AOC       A           Protein          99                 X-ray
3B80_A        3B80       A           Protein          99                 X-ray
3VF5_A        3VF5       A           Protein          99                 X-ray
2AVQ_A        2AVQ       A           Protein          99                 X-ray
1DW6_C        1DW6       C           Protein          99                 X-ray
1KZK_A        1KZK       A           Protein          99                 X-ray
2HS1_A        2HS1       A           Protein          99                 X-ray
1K6C_A        1K6C       A           Protein          99                 X-ray
1MTB_A        1MTB       A           Protein          99                 X-ray
4Q1X_A        4Q1X       A           Protein          99                 X-ray
4Q1W_A        4Q1W       A           Protein          99                 X-ray
3D1X_A        3D1X       A           Protein          99                 X-ray
2AVM_A        2AVM       A           Protein          99                 X-ray
3PWM_A        3PWM       A           Protein          99                 X-ray
3KT2_A        3KT2       A           Protein         203                 X-ray
1SDV_A        1SDV       A           Protein          99                 X-ray
3JVW_A        3JVW       A           Protein          99                 X-ray
3OY4_A        3OY4       A           Protein          99                 X-ray
1A94_A        1A94       A           Protein          99                 X-ray
2HS2_A        2HS2       A           Protein          99                 X-ray
       resolution                                   scopDomain
1W5V_A      1.800 Human immunodeficiency virus type 1 protease
2FDE_A      2.700 Human immunodeficiency virus type 1 protease
1AJV_A      2.000 Human immunodeficiency virus type 1 protease
2R38_A      1.810 Human immunodeficiency virus type 1 protease
2R3T_A      1.800 Human immunodeficiency virus type 1 protease
1HXB_A      2.300 Human immunodeficiency virus type 1 protease
1BV9_A      2.000 Human immunodeficiency virus type 1 protease
1AAQ_A      2.500 Human immunodeficiency virus type 1 protease
1AXA_A      2.000 Human immunodeficiency virus type 1 protease
1HVS_A      2.250 Human immunodeficiency virus type 1 protease
1ZP8_A      2.020 Human immunodeficiency virus type 1 protease
2QHC_A      2.802 Human immunodeficiency virus type 1 protease
1A8G_A      2.500 Human immunodeficiency virus type 1 protease
2O4L_A      1.330 Human immunodeficiency virus type 1 protease
5COK_A      1.801 Human immunodeficiency virus type 1 protease
1TCX_A      2.300 Human immunodeficiency virus type 1 protease
2Z54_A      2.310 Human immunodeficiency virus type 1 protease
1D4S_A      2.500 Human immunodeficiency virus type 1 protease
1BV7_A      2.000 Human immunodeficiency virus type 1 protease
1BWA_A      1.900 Human immunodeficiency virus type 1 protease
1A9M_A      2.300 Human immunodeficiency virus type 1 protease
2FLE_A      1.900                            automated matches
1ODY_A      2.000 Human immunodeficiency virus type 1 protease
1GNN_A      2.300 Human immunodeficiency virus type 1 protease
1GNM_A      2.300 Human immunodeficiency virus type 1 protease
5YRS_B      1.760                                         &lt;NA&gt;
5YRS_A      1.760                                         &lt;NA&gt;
1HEF_E      2.200 Human immunodeficiency virus type 1 protease
1ODX_A      2.000 Human immunodeficiency virus type 1 protease
4QGI_A      1.896 Human immunodeficiency virus type 1 protease
1BVE_A         NA Human immunodeficiency virus type 1 protease
2AZ8_A      2.000 Human immunodeficiency virus type 1 protease
1A30_A      2.000 Human immunodeficiency virus type 1 protease
6DH6_A      1.970 Human immunodeficiency virus type 1 protease
6DH0_A      1.899 Human immunodeficiency virus type 1 protease
2I4D_A      1.500 Human immunodeficiency virus type 1 protease
6OOS_A      1.900 Human immunodeficiency virus type 1 protease
1RL8_A      2.000 Human immunodeficiency virus type 1 protease
1ZSF_A      1.980 Human immunodeficiency virus type 1 protease
2Q64_A      2.500 Human immunodeficiency virus type 1 protease
6DH3_A      1.908 Human immunodeficiency virus type 1 protease
2NPH_A      1.650 Human immunodeficiency virus type 1 protease
2Q63_A      2.200 Human immunodeficiency virus type 1 protease
1LZQ_A      2.200 Human immunodeficiency virus type 1 protease
1FB7_A      2.600 Human immunodeficiency virus type 1 protease
1G6L_A      1.900 Human immunodeficiency virus type 1 protease
1HIV_A      2.000 Human immunodeficiency virus type 1 protease
6OOU_A      2.127 Human immunodeficiency virus type 1 protease
1HVC_A      1.800 Human immunodeficiency virus type 1 protease
2I4V_A      1.500 Human immunodeficiency virus type 1 protease
2AZ9_A      2.500 Human immunodeficiency virus type 1 protease
6OOT_A      1.822 Human immunodeficiency virus type 1 protease
2P3B_B      2.100 Human immunodeficiency virus type 1 protease
5KAO_A      1.800 Human immunodeficiency virus type 1 protease
2WL0_A      1.900                            automated matches
6OPT_A      1.960 Human immunodeficiency virus type 1 protease
1IZI_A      2.150 Human immunodeficiency virus type 1 protease
1MRX_A      2.000 Human immunodeficiency virus type 1 protease
2PYM_A      1.900 Human immunodeficiency virus type 1 protease
2PYN_A      1.850 Human immunodeficiency virus type 1 protease
1DMP_A      2.000 Human immunodeficiency virus type 1 protease
4K4P_A      2.310 Human immunodeficiency virus type 1 protease
1LV1_A      2.100 Human immunodeficiency virus type 1 protease
1AID_A      2.200 Human immunodeficiency virus type 1 protease
1ZBG_A      1.995 Human immunodeficiency virus type 1 protease
3TKG_A      1.360 Human immunodeficiency virus type 1 protease
5YOK_A      0.850 Human immunodeficiency virus type 1 protease
1FGC_C      1.900 Human immunodeficiency virus type 1 protease
3K4V_A      1.390 Human immunodeficiency virus type 1 protease
3KT5_A      1.801 Human immunodeficiency virus type 1 protease
4QLH_A      2.450                                         &lt;NA&gt;
2F3K_A      1.599                            automated matches
4Q5M_A      1.795 Human immunodeficiency virus type 1 protease
2AOC_A      1.300 Human immunodeficiency virus type 1 protease
3B80_A      1.500 Human immunodeficiency virus type 1 protease
3VF5_A      1.250 Human immunodeficiency virus type 1 protease
2AVQ_A      1.300 Human immunodeficiency virus type 1 protease
1DW6_C      1.880 Human immunodeficiency virus type 1 protease
1KZK_A      1.090 Human immunodeficiency virus type 1 protease
2HS1_A      0.840 Human immunodeficiency virus type 1 protease
1K6C_A      2.200 Human immunodeficiency virus type 1 protease
1MTB_A      2.500 Human immunodeficiency virus type 1 protease
4Q1X_A      1.900 Human immunodeficiency virus type 1 protease
4Q1W_A      1.850 Human immunodeficiency virus type 1 protease
3D1X_A      1.050 Human immunodeficiency virus type 1 protease
2AVM_A      1.100 Human immunodeficiency virus type 1 protease
3PWM_A      1.460 Human immunodeficiency virus type 1 protease
3KT2_A      1.651 Human immunodeficiency virus type 1 protease
1SDV_A      1.400 Human immunodeficiency virus type 1 protease
3JVW_A      1.800                            automated matches
3OY4_A      1.760 Human immunodeficiency virus type 1 protease
1A94_A      2.000 Human immunodeficiency virus type 1 protease
2HS2_A      1.220 Human immunodeficiency virus type 1 protease
                                     pfam              ligandId
1W5V_A Retroviral aspartyl protease (RVP)                   BE3
2FDE_A Retroviral aspartyl protease (RVP)             K (2),385
1AJV_A Retroviral aspartyl protease (RVP)                   NMB
2R38_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2R3T_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1HXB_A Retroviral aspartyl protease (RVP)                   ROC
1BV9_A Retroviral aspartyl protease (RVP)                   XV6
1AAQ_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1AXA_A Retroviral aspartyl protease (RVP)               U0E (2)
1HVS_A Retroviral aspartyl protease (RVP)                   A77
1ZP8_A Retroviral aspartyl protease (RVP)                   AB2
2QHC_A Retroviral aspartyl protease (RVP)                   BME
1A8G_A Retroviral aspartyl protease (RVP)                   2Z4
2O4L_A Retroviral aspartyl protease (RVP)            CL,TPV,GOL
5COK_A Retroviral aspartyl protease (RVP)                   52U
1TCX_A Retroviral aspartyl protease (RVP)                   IM1
2Z54_A Retroviral aspartyl protease (RVP)               BME,AB1
1D4S_A Retroviral aspartyl protease (RVP)                   TPV
1BV7_A Retroviral aspartyl protease (RVP)                   XV6
1BWA_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1A9M_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2FLE_A Retroviral aspartyl protease (RVP)                   GOL
1ODY_A Retroviral aspartyl protease (RVP)                   LP1
1GNN_A Retroviral aspartyl protease (RVP)               U0E (2)
1GNM_A Retroviral aspartyl protease (RVP)               U0E (2)
5YRS_B Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
5YRS_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1HEF_E Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1ODX_A Retroviral aspartyl protease (RVP)                   0E8
4QGI_A Retroviral aspartyl protease (RVP)               ROC,GOL
1BVE_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2AZ8_A Retroviral aspartyl protease (RVP)                   3TL
1A30_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
6DH6_A Retroviral aspartyl protease (RVP)           SO4 (3),017
6DH0_A Retroviral aspartyl protease (RVP)                   017
2I4D_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
6OOS_A Retroviral aspartyl protease (RVP)           SO4 (4),017
1RL8_A Retroviral aspartyl protease (RVP)                   RIT
1ZSF_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2Q64_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
6DH3_A Retroviral aspartyl protease (RVP)           017,SO4 (2)
2NPH_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2Q63_A Retroviral aspartyl protease (RVP)                   1UN
1LZQ_A Retroviral aspartyl protease (RVP)               0ZQ,BME
1FB7_A Retroviral aspartyl protease (RVP)                   ROC
1G6L_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1HIV_A Retroviral aspartyl protease (RVP)                   1ZK
6OOU_A Retroviral aspartyl protease (RVP)           017,SO4 (2)
1HVC_A Retroviral aspartyl protease (RVP)                   A79
2I4V_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2AZ9_A Retroviral aspartyl protease (RVP)                   3TL
6OOT_A Retroviral aspartyl protease (RVP)                   SO4
2P3B_B Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
5KAO_A Retroviral aspartyl protease (RVP)                   G43
2WL0_A Retroviral aspartyl protease (RVP)                   5AH
6OPT_A Retroviral aspartyl protease (RVP)           017,SO4 (2)
1IZI_A Retroviral aspartyl protease (RVP)                    CL
1MRX_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2PYM_A Retroviral aspartyl protease (RVP)                   1UN
2PYN_A Retroviral aspartyl protease (RVP)                   1UN
1DMP_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
4K4P_A Retroviral aspartyl protease (RVP)                   NO3
1LV1_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1AID_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1ZBG_A Retroviral aspartyl protease (RVP)                CL (2)
3TKG_A Retroviral aspartyl protease (RVP)                CL,GOL
5YOK_A Retroviral aspartyl protease (RVP)                   8Z0
1FGC_C Retroviral aspartyl protease (RVP)                   2NC
3K4V_A Retroviral aspartyl protease (RVP)               DMS (2)
3KT5_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
4QLH_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
2F3K_A Retroviral aspartyl protease (RVP)           PO4 (3),RO1
4Q5M_A Retroviral aspartyl protease (RVP)                   ROC
2AOC_A Retroviral aspartyl protease (RVP) UNX,NA,CL,DMS (3),GOL
3B80_A Retroviral aspartyl protease (RVP)                 NA,CL
3VF5_A Retroviral aspartyl protease (RVP)     NA,CL (2),ACT (2)
2AVQ_A Retroviral aspartyl protease (RVP)               DMS,GOL
1DW6_C Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1KZK_A Retroviral aspartyl protease (RVP)    JE2,CL (2),EDO (4)
2HS1_A Retroviral aspartyl protease (RVP)            CL (3),017
1K6C_A Retroviral aspartyl protease (RVP)               ACT (2)
1MTB_A Retroviral aspartyl protease (RVP)                   ROC
4Q1X_A Retroviral aspartyl protease (RVP)               017,GOL
4Q1W_A Retroviral aspartyl protease (RVP)           PO4 (3),017
3D1X_A Retroviral aspartyl protease (RVP)            CL,ROC,GOL
2AVM_A Retroviral aspartyl protease (RVP)                   GOL
3PWM_A Retroviral aspartyl protease (RVP)         CL,ACT (2),NA
3KT2_A Retroviral aspartyl protease (RVP)                  &lt;NA&gt;
1SDV_A Retroviral aspartyl protease (RVP)                    CL
3JVW_A Retroviral aspartyl protease (RVP)                   DMP
3OY4_A Retroviral aspartyl protease (RVP)       PO4 (2),ACT (3)
1A94_A Retroviral aspartyl protease (RVP)                   0Q4
2HS2_A Retroviral aspartyl protease (RVP)            CL (2),017
                                                                                                                                                                                                                            ligandName
1W5V_A                                                                                                                                                                 N,N-[2,5-O-DI-3-FLUORO-BENZYL-GLUCARYL]-DI-[1-AMINO-INDAN-2-OL]
2FDE_A                             POTASSIUM ION (2),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL [(1S,2R)-3-[(1,3-BENZODIOXOL-5-YLSULFONYL)(ISOBUTYL)AMINO]-2-HYDROXY-1-{4-[(2-METHYL-1,3-THIAZOL-4-YL)METHOXY]BENZYL}PROPYL]CARBAMATE
1AJV_A                                                                                                                                                      2,7-DIBENZYL-1,1-DIOXO-3,6-BIS-PHENOXYMETHYL-[1,2,7]THIADIAZEPANE-4,5-DIOL
2R38_A                                                                                                                                                                                                                            &lt;NA&gt;
2R3T_A                                                                                                                                                                                                                            &lt;NA&gt;
1HXB_A                                        (2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide
1BV9_A                                                        [4R-(4ALPHA,5ALPHA,6BETA,7BETA)]-3,3'-[[TETRAHYDRO-5,6-DIHYDROXY-2-OXO-4,7-BIS(PHENYLMETHYL)-1H-1,3-DIAZEPINE-1,3(2H)-D\nIYL] BIS(METHYLENE)]BIS[N-2-THIAZOLYLBENZAMIDE]
1AAQ_A                                                                                                                                                                                                                            &lt;NA&gt;
1AXA_A                                                                                                                N-[[1-[N-ACETAMIDYL]-[1-CYCLOHEXYLMETHYL-2-HYDROXY-4-ISOPROPYL]-BUT-4-YL]-CARBONYL]-GLUTAMINYL-ARGINYL-AMIDE (2)
1HVS_A                                              N-{1-BENZYL-(2R,3S)-2,3-DIHYDROXY-4-[3-METHYL-2-(3-METHYL-3-PYRIDIN-2-YLMETHYL-UREIDO)-BUTYRYLAMINO]-5-PHENYL-PENTYL}-3-METHYL-2-(3-METHYL-3-PYRIDIN-2-YLMETHYL-UREIDO)-BUTYRAMIDE
1ZP8_A                                                            [1-((1S,2R)-1-BENZYL-2-HYDROXY-3-{ISOBUTYL[(4-METHOXYPHENYL)SULFONYL]AMINO}PROPYL)-1H-1,2,3-TRIAZOL-4-YL]METHYL (1R,2R)-2-HYDROXY-2,3-DIHYDRO-1H-INDEN-1-YLCARBAMATE
2QHC_A                                                                                                                                                                                                            BETA-MERCAPTOETHANOL
1A8G_A                            benzyl [(1R)-1-({(1S,2S,3S)-1-benzyl-2-hydroxy-4-({(1S)-1-[(2-hydroxy-4-methoxybenzyl)carbamoyl]-2-methylpropyl}amino)-3-[(4-methoxybenzyl)amino]-4-oxobutyl}carbamoyl)-2,2-dimethylpropyl]carbamate
2O4L_A                                                                CHLORIDE ION,N-(3-{(1R)-1-[(6R)-4-HYDROXY-2-OXO-6-PHENETHYL-6-PROPYL-5,6-DIHYDRO-2H-PYRAN-3-YL]PROPYL}PHENYL)-5-(TRIFLUOROMETHYL)-2-PYRIDINESULFONAMIDE,GLYCEROL
5COK_A                                                                           (3aS,4S,7aR)-hexahydro-4H-furo[2,3-b]pyran-4-yl\n[(2S,3R)-3-hydroxy-4-{[(4-methoxyphenyl)sulfonyl](2-methylpropyl)amino}-1-phenylbutan-2-yl]carbamate
1TCX_A                                                                                                 (2R,4S,5S,1'S)-2-PHENYLMETHYL-4-HYDROXY-5-(TERT-BUTOXYCARBONYL)AMINO-6-PHENYL HEXANOYL-N-(1'-IMIDAZO-2-YL)-2'-METHYLPROPANAMIDE
2Z54_A                                                                    BETA-MERCAPTOETHANOL,N-{1-BENZYL-4-[2-(2,6-DIMETHYL-PHENOXY)-ACETYLAMINO]-3-HYDROXY-5-PHENYL-PENTYL}-3-METHYL-2-(2-OXO-TETRAHYDRO-PYRIMIDIN-1-YL)-BUTYRAMIDE
1D4S_A                                                                                      N-(3-{(1R)-1-[(6R)-4-HYDROXY-2-OXO-6-PHENETHYL-6-PROPYL-5,6-DIHYDRO-2H-PYRAN-3-YL]PROPYL}PHENYL)-5-(TRIFLUOROMETHYL)-2-PYRIDINESULFONAMIDE
1BV7_A                                                        [4R-(4ALPHA,5ALPHA,6BETA,7BETA)]-3,3'-[[TETRAHYDRO-5,6-DIHYDROXY-2-OXO-4,7-BIS(PHENYLMETHYL)-1H-1,3-DIAZEPINE-1,3(2H)-D\nIYL] BIS(METHYLENE)]BIS[N-2-THIAZOLYLBENZAMIDE]
1BWA_A                                                                                                                                                                                                                            &lt;NA&gt;
1A9M_A                                                                                                                                                                                                                            &lt;NA&gt;
2FLE_A                                                                                                                                                                                                                        GLYCEROL
1ODY_A                                                 4-[2-(2-ACETYLAMINO-3-NAPHTALEN-1-YL-PROPIONYLAMINO)-4-METHYL-PENTANOYLAMINO]-3-HYDROXY-6-METHYL-HEPTANOIC ACID [1-(1-CARBAMOYL-2-NAPHTHALEN-1-YL-ETHYLCARBAMOYL)-PROPYL]-AMIDE
1GNN_A                                                                                                                N-[[1-[N-ACETAMIDYL]-[1-CYCLOHEXYLMETHYL-2-HYDROXY-4-ISOPROPYL]-BUT-4-YL]-CARBONYL]-GLUTAMINYL-ARGINYL-AMIDE (2)
1GNM_A                                                                                                                N-[[1-[N-ACETAMIDYL]-[1-CYCLOHEXYLMETHYL-2-HYDROXY-4-ISOPROPYL]-BUT-4-YL]-CARBONYL]-GLUTAMINYL-ARGINYL-AMIDE (2)
5YRS_B                                                                                                                                                                                                                            &lt;NA&gt;
5YRS_A                                                                                                                                                                                                                            &lt;NA&gt;
1HEF_E                                                                                                                                                                                                                            &lt;NA&gt;
1ODX_A                                                                                                                                                 di-tert-butyl {iminobis[(2S,3S)-3-hydroxy-1-phenylbutane-4,2-diyl]}biscarbamate
4QGI_A                               (2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide,GLYCEROL
1BVE_A                                                                                                                                                                                                                            &lt;NA&gt;
2AZ8_A                                          benzyl [(1S,4S,7S,8R,9R,10S,13S,16S)-7,10-dibenzyl-8,9-dihydroxy-1,16-dimethyl-4,13-bis(1-methylethyl)-2,5,12,15,18-pentaoxo-20-phenyl-19-oxa-3,6,11,14,17-pentaazaicos-1-yl]carbamate
1A30_A                                                                                                                                                                                                                            &lt;NA&gt;
6DH6_A                                                                               SULFATE ION (3),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
6DH0_A                                                                                               (3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
2I4D_A                                                                                                                                                                                                                            &lt;NA&gt;
6OOS_A                                                                               SULFATE ION (4),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
1RL8_A                                                                                                                                                                                                                       RITONAVIR
1ZSF_A                                                                                                                                                                                                                            &lt;NA&gt;
2Q64_A                                                                                                                                                                                                                            &lt;NA&gt;
6DH3_A                                                                               (3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE,SULFATE ION (2)
2NPH_A                                                                                                                                                                                                                            &lt;NA&gt;
2Q63_A                                                                                              2-[2-HYDROXY-3-(3-HYDROXY-2-METHYL-BENZOYLAMINO)-4-PHENYL SULFANYL-BUTYL]-DECAHYDRO-ISOQUINOLINE-3-CARBOXYLIC ACID TERT-BUTYLAMIDE
1LZQ_A                                                                                                  N-{(3S)-3-[(tert-butoxycarbonyl)amino]-4-phenylbutyl}-L-phenylalanyl-L-alpha-glutamyl-L-phenylalaninamide,BETA-MERCAPTOETHANOL
1FB7_A                                        (2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide
1G6L_A                                                                                                                                                                                                                            &lt;NA&gt;
1HIV_A 4-[(2R)-3-{[(1S,2S,3R,4S)-1-(cyclohexylmethyl)-2,3-dihydroxy-5-methyl-4-({(1S,2R)-2-methyl-1-[(pyridin-2-ylmethyl)carba\nmoyl]butyl}carbamoyl)hexyl]amino}-2-{[(naphthalen-1-yloxy)acetyl]amino}-3-oxopropyl]-1H-imidazol-3-ium
6OOU_A                                                                               (3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE,SULFATE ION (2)
1HVC_A                                              N-{1-BENZYL-(2S,3S)-2,3-DIHYDROXY-4-[3-METHYL-2-(3-METHYL-3-PYRIDIN-2-YLMETHYL-UREIDO)-BUTYRYLAMINO]-5-PHENYL-PENTYL}-3-METHYL-2-(3-METHYL-3-PYRIDIN-2-YLMETHYL-UREIDO)-BUTYRAMIDE
2I4V_A                                                                                                                                                                                                                            &lt;NA&gt;
2AZ9_A                                          benzyl [(1S,4S,7S,8R,9R,10S,13S,16S)-7,10-dibenzyl-8,9-dihydroxy-1,16-dimethyl-4,13-bis(1-methylethyl)-2,5,12,15,18-pentaoxo-20-phenyl-19-oxa-3,6,11,14,17-pentaazaicos-1-yl]carbamate
6OOT_A                                                                                                                                                                                                                     SULFATE ION
2P3B_B                                                                                                                                                                                                                            &lt;NA&gt;
5KAO_A       [(3~{a}~{S},4~{R},6~{a}~{R})-2,3,3~{a},4,5,6~{a}-hexahydrofuro[2,3-b]furan-4-yl] ~{N}-[(2~{S},3~{R})-1-(3-chloranyl-4-methoxy-phenyl)-4-[(4-methoxyphenyl)sulfonyl-(2-methylpropyl)amino]-3-oxidanyl-butan-2-yl]carbamate
2WL0_A                                         METHYL [(1S)-1-({2-[(3S)-3-BENZYL-3-HYDROXY-4-{[(1S,2R)-2-HYDROXY-2,3-DIHYDRO-1H-INDEN-1-YL]AMINO}-4-OXOBUTYL]-2-(4-PYRIDIN-2-YLBENZYL)HYDRAZINO}CARBONYL)-2,2-DIMETHYLPROPYL]CARBAMATE
6OPT_A                                                                               (3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE,SULFATE ION (2)
1IZI_A                                                                                                                                                                                                                    CHLORIDE ION
1MRX_A                                                                                                                                                                                                                            &lt;NA&gt;
2PYM_A                                                                                              2-[2-HYDROXY-3-(3-HYDROXY-2-METHYL-BENZOYLAMINO)-4-PHENYL SULFANYL-BUTYL]-DECAHYDRO-ISOQUINOLINE-3-CARBOXYLIC ACID TERT-BUTYLAMIDE
2PYN_A                                                                                              2-[2-HYDROXY-3-(3-HYDROXY-2-METHYL-BENZOYLAMINO)-4-PHENYL SULFANYL-BUTYL]-DECAHYDRO-ISOQUINOLINE-3-CARBOXYLIC ACID TERT-BUTYLAMIDE
1DMP_A                                                                                                                                                                                                                            &lt;NA&gt;
4K4P_A                                                                                                                                                                                                                     NITRATE ION
1LV1_A                                                                                                                                                                                                                            &lt;NA&gt;
1AID_A                                                                                                                                                                                                                            &lt;NA&gt;
1ZBG_A                                                                                                                                                                                                                CHLORIDE ION (2)
3TKG_A                                                                                                                                                                                                           CHLORIDE ION,GLYCEROL
5YOK_A          (4R)-N-[(2,6-dimethylphenyl)methyl]-3-[(2S,3S)-3-[[(2S)-2-[(7-methoxy-1-benzofuran-2-yl)carbonylamino]-2-[(3R)-oxolan-3\n-yl]ethanoyl]amino]-2-oxidanyl-4-phenyl-butanoyl]-5,5-dimethyl-1,3-thiazolidine-4-carboxamide
1FGC_C                                                                                                    N-{(2S)-2-[(N-acetyl-L-threonyl-L-isoleucyl)amino]hexyl}-L-norleucyl-L-glutaminyl-N~5~-[amino(iminio)methyl]-L-ornithinamide
3K4V_A                                                                                                                                                                                                          DIMETHYL SULFOXIDE (2)
3KT5_A                                                                                                                                                                                                                            &lt;NA&gt;
4QLH_A                                                                                                                                                                                                                            &lt;NA&gt;
2F3K_A                                                 PHOSPHATE ION (3),(3S,4AS,8AS)-N-(TERT-BUTYL)-2-[(3S)-3-({3-(METHYLSULFONYL)-N-[(PYRIDIN-3-YLOXY)ACETYL]-L-VALYL}AMINO)-2-OXO-4-PHENYLBUTYL]DECAHYDROISOQUINOLINE-3-CARBOXAMIDE
4Q5M_A                                        (2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide
2AOC_A                                                                                                                                                     UNKNOWN ATOM OR ION,SODIUM ION,CHLORIDE ION,DIMETHYL SULFOXIDE (3),GLYCEROL
3B80_A                                                                                                                                                                                                         SODIUM ION,CHLORIDE ION
3VF5_A                                                                                                                                                                                     SODIUM ION,CHLORIDE ION (2),ACETATE ION (2)
2AVQ_A                                                                                                                                                                                                     DIMETHYL SULFOXIDE,GLYCEROL
1DW6_C                                                                                                                                                                                                                            &lt;NA&gt;
1KZK_A                                             (4R)-3-{(2S,3S)-2-hydroxy-3-[(3-hydroxy-2-methylbenzoyl)amino]-4-phenylbutanoyl}-5,5-dimethyl-N-(2-methylbenzyl)-1,3-thiazolidine-4-carboxamide,CHLORIDE ION (2),1,2-ETHANEDIOL (4)
2HS1_A                                                                              CHLORIDE ION (3),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
1K6C_A                                                                                                                                                                                                                 ACETATE ION (2)
1MTB_A                                        (2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide
4Q1X_A                                                                                      (3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE,GLYCEROL
4Q1W_A                                                                             PHOSPHATE ION (3),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
3D1X_A                  CHLORIDE ION,(2S)-N-[(2S,3R)-4-[(2S,3S,4aS,8aS)-3-(tert-butylcarbamoyl)-3,4,4a,5,6,7,8,8a-octahydro-1H-isoquinolin-2-yl]-3-hydroxy-1\n-phenyl-butan-2-yl]-2-(quinolin-2-ylcarbonylamino)butanediamide,GLYCEROL
2AVM_A                                                                                                                                                                                                                        GLYCEROL
3PWM_A                                                                                                                                                                                         CHLORIDE ION,ACETATE ION (2),SODIUM ION
3KT2_A                                                                                                                                                                                                                            &lt;NA&gt;
1SDV_A                                                                                                                                                                                                                    CHLORIDE ION
3JVW_A                                                                          [4-R-(-4-ALPHA,5-ALPHA,6-BETA,7-BETA)]-HEXAHYDRO-5,6-BIS(HYDROXY)-[1,3-BIS([4-HYDROXYMETHYL-PHENYL]METHYL)-4,7-BIS(PHEN\nYLMETHYL)]-2H-1,3-DIAZEPINONE
3OY4_A                                                                                                                                                                                               PHOSPHATE ION (2),ACETATE ION (3)
1A94_A                                                                                      N-[(2R)-2-({N~5~-[amino(iminio)methyl]-L-ornithyl-L-valyl}amino)-4-methylpentyl]-L-phenylalanyl-L-alpha-glutamyl-L-alanyl-L-norleucinamide
2HS2_A                                                                              CHLORIDE ION (2),(3R,3AS,6AR)-HEXAHYDROFURO[2,3-B]FURAN-3-YL(1S,2R)-3-[[(4-AMINOPHENYL)SULFONYL](ISOBUTYL)AMINO]-1-BENZYL-2-HYDROXYPROPYLCARBAMATE
                                                                       source
1W5V_A                                           Human immunodeficiency virus
2FDE_A                                         Human immunodeficiency virus 1
1AJV_A                                         Human immunodeficiency virus 1
2R38_A                                         Human immunodeficiency virus 1
2R3T_A                                         Human immunodeficiency virus 1
1HXB_A                         Human immunodeficiency virus type 1 (CLONE 12)
1BV9_A                                         Human immunodeficiency virus 1
1AAQ_A                                         Human immunodeficiency virus 1
1AXA_A                                         Human immunodeficiency virus 1
1HVS_A                                         Human immunodeficiency virus 1
1ZP8_A                                         Human immunodeficiency virus 1
2QHC_A                                         Human immunodeficiency virus 1
1A8G_A                                         Human immunodeficiency virus 1
2O4L_A                                         Human immunodeficiency virus 1
5COK_A                                         Human immunodeficiency virus 1
1TCX_A                                         Human immunodeficiency virus 1
2Z54_A                      Human immunodeficiency virus type 1 (BH5 ISOLATE)
1D4S_A                                         Human immunodeficiency virus 1
1BV7_A                                         Human immunodeficiency virus 1
1BWA_A                                         Human immunodeficiency virus 1
1A9M_A                                         Human immunodeficiency virus 1
2FLE_A                                         Human immunodeficiency virus 1
1ODY_A                                                        HIV-1 M:B_HXB2R
1GNN_A                                         Human immunodeficiency virus 1
1GNM_A                                         Human immunodeficiency virus 1
5YRS_B                                                        HIV-1 M:B_HXB2R
5YRS_A                                                        HIV-1 M:B_HXB2R
1HEF_E                                         Human immunodeficiency virus 1
1ODX_A                                         Human immunodeficiency virus 1
4QGI_A                     Human immunodeficiency virus type 1 lw12.3 isolate
1BVE_A                                         Human immunodeficiency virus 1
2AZ8_A                                         Human immunodeficiency virus 1
1A30_A                                         Human immunodeficiency virus 1
6DH6_A                                         Human immunodeficiency virus 1
6DH0_A                                         Human immunodeficiency virus 1
2I4D_A                                         Human immunodeficiency virus 1
6OOS_A                                         Human immunodeficiency virus 1
1RL8_A                                         Human immunodeficiency virus 1
1ZSF_A                                         Human immunodeficiency virus 1
2Q64_A                                         Human immunodeficiency virus 1
6DH3_A                                         Human immunodeficiency virus 1
2NPH_A                                         Human immunodeficiency virus 1
2Q63_A                                         Human immunodeficiency virus 1
1LZQ_A                                         Human immunodeficiency virus 1
1FB7_A                                         Human immunodeficiency virus 1
1G6L_A                                         Human immunodeficiency virus 1
1HIV_A                                         Human immunodeficiency virus 1
6OOU_A                                         Human immunodeficiency virus 1
1HVC_A                                         Human immunodeficiency virus 1
2I4V_A                                         Human immunodeficiency virus 1
2AZ9_A                                         Human immunodeficiency virus 1
6OOT_A                                         Human immunodeficiency virus 1
2P3B_B                                         Human immunodeficiency virus 1
5KAO_A                                         Human immunodeficiency virus 1
2WL0_A               Human immunodeficiency virus type 1 (Z2/CDC-Z34 ISOLATE)
6OPT_A                                         Human immunodeficiency virus 1
1IZI_A                                         Human immunodeficiency virus 1
1MRX_A                                         Human immunodeficiency virus 1
2PYM_A                                         Human immunodeficiency virus 1
2PYN_A                                         Human immunodeficiency virus 1
1DMP_A                                         Human immunodeficiency virus 1
4K4P_A               Human immunodeficiency virus type 1 (Z2/CDC-Z34 ISOLATE)
1LV1_A                                         Human immunodeficiency virus 1
1AID_A                                         Human immunodeficiency virus 1
1ZBG_A                                         Human immunodeficiency virus 1
3TKG_A                      Human immunodeficiency virus type 1 (BRU ISOLATE)
5YOK_A                                         Human immunodeficiency virus 1
1FGC_C                                         Human immunodeficiency virus 1
3K4V_A                               Human immunodeficiency virus type 1 BH10
3KT5_A                                                        HIV-1 M:B_HXB2R
4QLH_A                     Human immunodeficiency virus 1/synthetic construct
2F3K_A                                         Human immunodeficiency virus 1
4Q5M_A                                                        HIV-1 M:B_HXB2R
2AOC_A                      Human immunodeficiency virus type 1 (BH5 ISOLATE)
3B80_A                               Human immunodeficiency virus type 1 BH10
3VF5_A                      Human immunodeficiency virus type 1 (BRU ISOLATE)
2AVQ_A                      Human immunodeficiency virus type 1 (BH5 ISOLATE)
1DW6_C                                         Human immunodeficiency virus 1
1KZK_A                                           Human immunodeficiency virus
2HS1_A                                         Human immunodeficiency virus 1
1K6C_A                                         Human immunodeficiency virus 1
1MTB_A                                         Human immunodeficiency virus 1
4Q1X_A                                         Human immunodeficiency virus 1
4Q1W_A                                         Human immunodeficiency virus 1
3D1X_A NA/Human immunodeficiency virus type 1 group M subtype B (isolate BH5)
2AVM_A                                         Human immunodeficiency virus 1
3PWM_A                                         Human immunodeficiency virus 1
3KT2_A                                                        HIV-1 M:B_HXB2R
1SDV_A                                         Human immunodeficiency virus 1
3JVW_A                      Human immunodeficiency virus type 1 (BRU ISOLATE)
3OY4_A                                                     HIV-1 M:B_ARV2/SF2
1A94_A                                         Human immunodeficiency virus 1
2HS2_A                                         Human immunodeficiency virus 1
                                                                                                                                                                                                       structureTitle
1W5V_A                                                                                                                           HIV-1 protease in complex with fluoro substituted diol-based C2- symmetric inhibitor
2FDE_A                                                                                                                                                                       Wild type HIV protease bound with GW0385
1AJV_A                                                                                                                                           HIV-1 PROTEASE IN COMPLEX WITH THE CYCLIC SULFAMIDE INHIBITOR AHA006
2R38_A                                                                                                                   I84V HIV-1 protease mutant in complex with a carbamoyl decorated pyrrolidine-based inhibitor
2R3T_A                                                                                                                   I50V HIV-1 protease mutant in complex with a carbamoyl decorated pyrrolidine-based inhibitor
1HXB_A                                                                                                                                                                     HIV-1 proteinase complexed with RO 31-8959
1BV9_A                                                                                                                                           HIV-1 PROTEASE (I84V) COMPLEXED WITH XV638 OF DUPONT PHARMACEUTICALS
1AAQ_A                           HYDROXYETHYLENE ISOSTERE INHIBITORS OF HUMAN IMMUNODEFICIENCY VIRUS-1 PROTEASE: STRUCTURE-ACTIVITY ANALYSIS USING ENZYME KINETICS, X-RAY CRYSTALLOGRAPHY, AND INFECTED T-CELL ASSAYS
1AXA_A                                                                                       ACTIVE-SITE MOBILITY IN HUMAN IMMUNODEFICIENCY VIRUS TYPE 1 PROTEASE AS DEMONSTRATED BY CRYSTAL STRUCTURE OF A28S MUTANT
1HVS_A                                                                                          STRUCTURAL BASIS OF DRUG RESISTANCE FOR THE V82A MUTANT OF HIV-1 PROTEASE: BACKBONE FLEXIBILITY AND SUBSITE REPACKING
1ZP8_A                                                                                                                                                                               HIV Protease with inhibitor AB-2
2QHC_A                                                                                                                   The Influence of I47A Mutation on Reduced Susceptibility to the Protease Inhibitor Lopinavir
1A8G_A                                                                                                                                                                      HIV-1 PROTEASE IN COMPLEX WITH SDZ283-910
2O4L_A                                                                                                                                     Crystal Structure of HIV-1 Protease (Q7K, I50V) in Complex with Tipranavir
5COK_A                                                                                                                                   X-ray crystal structure of wild type HIV-1 protease in complex with GRL-0476
1TCX_A                                                                                                                                                   HIV TRIPLE MUTANT PROTEASE COMPLEXED WITH INHIBITOR SB203386
2Z54_A                                                                                                                   The Influence of I47A Mutation on Reduced Susceptibility to the Protease Inhibitor Lopinavir
1D4S_A                                                                                                                                                      HIV-1 PROTEASE V82F/I84V DOUBLE MUTANT/TIPRANAVIR COMPLEX
1BV7_A                                              COUNTERACTING HIV-1 PROTEASE DRUG RESISTANCE: STRUCTURAL ANALYSIS OF MUTANT PROTEASES COMPLEXED WITH XV638 AND SD146, CYCLIC UREA AMIDES WITH BROAD SPECIFICITIES
1BWA_A                                                                                                                        HIV-1 PROTEASE (V82F/I84V) DOUBLE MUTANT COMPLEXED WITH XV638 OF DUPONT PHARMACEUTICALS
1A9M_A                                                                                                                                    G48H MUTANT OF HIV-1 PROTEASE IN COMPLEX WITH A PEPTIDIC INHIBITOR U-89360E
2FLE_A                                                                                                                            Structural analysis of asymmetric inhibitor bound to the HIV-1 Protease V82A mutant
1ODY_A                                                                                                                                                              HIV-1 PROTEASE COMPLEXED WITH AN INHIBITOR LP-130
1GNN_A                                                                                                                    HIV-1 PROTEASE MUTANT WITH VAL 82 REPLACED BY ASN (V82N) COMPLEXED WITH U89360E (INHIBITOR)
1GNM_A                                                                                                                    HIV-1 PROTEASE MUTANT WITH VAL 82 REPLACED BY ASP (V82D) COMPLEXED WITH U89360E (INHIBITOR)
5YRS_B                                                                                      X-ray Snapshot of HIV-1 Protease in Action: Observation of Tetrahedral Intermediate and Its SIHB with Catalytic Aspartate
5YRS_A                                                                                      X-ray Snapshot of HIV-1 Protease in Action: Observation of Tetrahedral Intermediate and Its SIHB with Catalytic Aspartate
1HEF_E The crystal structures at 2.2 angstroms resolution of hydroxyethylene-based inhibitors bound to human immunodeficiency virus type 1 protease show that the inhibitors are present in two distinct orientations
1ODX_A                                                                                                                                                                             HIV-1 Proteinase mutant A71T, V82A
4QGI_A                                                                                                                         X-ray crystal structure of HIV-1 protease variant G48T/L89M in complex with Saquinavir
1BVE_A                                                                                                                                                  HIV-1 PROTEASE-DMP323 COMPLEX IN SOLUTION, NMR, 28 STRUCTURES
2AZ8_A                                                                                                                                                           HIV-1 Protease NL4-3 in complex with inhibitor, TL-3
1A30_A                                                                                                                                                           HIV-1 PROTEASE COMPLEXED WITH A TRIPEPTIDE INHIBITOR
6DH6_A                                                                                                                                Crystal structure of HIV-1 Protease NL4-3 I50V Mutant in complex with darunavir
6DH0_A                                                                                                                                Crystal structure of HIV-1 Protease NL4-3 I84V Mutant in complex with darunavir
2I4D_A                                                                                                                                                            Crystal structure of WT HIV-1 protease with GS-8373
6OOS_A                                                                                                                                                     HIV-1 Protease NL4-3 L90M Mutant in complex with darunavir
1RL8_A                                                                                                             Crystal structure of the complex of resistant strain of hiv-1 protease(v82a mutant) with ritonavir
1ZSF_A                                                                                                           Crystal Structure of Complex of a Hydroxyethylamine Inhibitor with HIV-1 Protease at 2.0A Resolution
2Q64_A                                                                                                                                                                     HIV-1 PR mutant in complex with nelfinavir
6DH3_A                                                                                                                                Crystal structure of HIV-1 Protease NL4-3 V82I Mutant in complex with darunavir
2NPH_A                                                                                                                                                     Crystal structure of HIV1 protease in situ product complex
2Q63_A                                                                                                                                                                     HIV-1 PR mutant in complex with nelfinavir
1LZQ_A                                               Crystal structure of the complex of mutant HIV-1 protease (A71V, V82T, I84V) with an ethylenamine peptidomimetic inhibitor BOC-PHE-PSI[CH2CH2NH]-PHE-GLU-PHE-NH2
1FB7_A                                                                              CRYSTAL STRUCTURE OF AN IN VIVO HIV-1 PROTEASE MUTANT IN COMPLEX WITH SAQUINAVIR: INSIGHTS INTO THE MECHANISMS OF DRUG RESISTANCE
1G6L_A                                                                                                                                                              1.9A CRYSTAL STRUCTURE OF TETHERED HIV-1 PROTEASE
1HIV_A                                                                              CRYSTAL STRUCTURE OF A COMPLEX OF HIV-1 PROTEASE WITH A DIHYDROETHYLENE-CONTAINING INHIBITOR: COMPARISONS WITH MOLECULAR MODELING
6OOU_A                                                                                                                                Crystal structure of HIV-1 Protease NL4-3 L89V Mutant in complex with darunavir
1HVC_A                                                                                                                            CRYSTAL STRUCTURE OF A TETHERED DIMER OF HIV-1 PROTEASE COMPLEXED WITH AN INHIBITOR
2I4V_A                                                                                                                                                                          HIV-1 protease I84V, L90M with TMC126
2AZ9_A                                                                                                                                                                                 HIV-1 Protease NL4-3 1X mutant
6OOT_A                                                                                                                                               HIV-1 Protease NL4-3 L89V, L90M Mutant in complex with darunavir
2P3B_B                                                                                                                        Crystal Structure of the subtype B wild type HIV protease complexed with TL-3 inhibitor
5KAO_A                                                                                                                                        Crystal structure of wild type HIV-1 protease in complex with GRL-10413
2WL0_A                                                                              HIV-1 Protease Inhibitors Containing a Tertiary Alcohol in the Transition-State Mimic with Improved Cell-Based Antiviral Activity
6OPT_A                                                                                                                                               HIV-1 Protease NL4-3 V82F, I84V Mutant in complex with darunavir
1IZI_A                                                                                                       Inhibitor of HIV protease with unusual binding mode potently inhibiting multi-resistant protease mutants
1MRX_A                                                                                                                             Structure of HIV protease (Mutant Q7K L33I L63I V82F I84V ) complexed with KNI-577
2PYM_A                                                                                                                                                                     HIV-1 PR mutant in complex with nelfinavir
2PYN_A                                                                                                                                                                     HIV-1 PR mutant in complex with nelfinavir
1DMP_A                                                                                                                                                                            STRUCTURE OF HIV-1 PROTEASE COMPLEX
4K4P_A                                                                                                                                                                            TL-3 inhibited Trp6Ala HIV Protease
1LV1_A                                                                                                         Crystal Structure Analysis of the non-active site mutant of tethered HIV-1 protease to 2.1A resolution
1AID_A                                                                                          STRUCTURE OF A NON-PEPTIDE INHIBITOR COMPLEXED WITH HIV-1 PROTEASE: DEVELOPING A CYCLE OF STRUCTURE-BASED DRUG DESIGN
1ZBG_A                                        Crystal structure of a complex of mutant hiv-1 protease (A71V, V82T, I84V) with a hydroxyethylamine peptidomimetic inhibitor BOC-PHE-PSI[R-CH(OH)CH2NH]-PHE-GLU-PHE-NH2
3TKG_A                                                                                                                                           crystal structure of HIV model protease precursor/saquinavir complex
5YOK_A                                                                                                                                                 Structure of HIV-1 Protease in Complex with Inhibitor KNI-1657
1FGC_C                                                      STRUCTURAL IMPLICATIONS OF DRUG RESISTANT MUTANTS OF HIV-1 PROTEASE: HIGH RESOLUTION CRYSTAL STRUCTURES OF THE MUTANT PROTEASE/SUBSTRATE ANALOG COMPLEXES
3K4V_A                                                                                                            New crystal form of HIV-1 Protease/Saquinavir structure reveals carbamylation of N-terminal proline
3KT5_A                                                                                                                                                                Crystal Structure of N88S mutant HIV-1 Protease
4QLH_A                                                                                                                                                 Crystal structure of drug resistant V82S/V1082S HIV-1 Protease
2F3K_A                                                                                                      Substrate envelope and drug resistance: crystal structure of r01 in complex with wild-type hiv-1 protease
4Q5M_A                                                                                                                                                          D30N tethered HIV-1 protease dimer/saquinavir complex
2AOC_A                                                                                                                         Crystal structure analysis of HIV-1 protease mutant I84V with a substrate analog P2-NC
3B80_A                                                                                                                                   HIV-1 protease mutant I54V complexed with gem-diol-amine intermediate NLLTQI
3VF5_A                                                                                                                               Crystal Structure of HIV-1 Protease Mutant I47V with novel P1'-Ligands GRL-02031
2AVQ_A                                                         Kinetics, stability, and structural changes in high resolution crystal structures of HIV-1 protease with drug resistant mutations L24I, I50V, AND G73S
1DW6_C                                                                                                                                    Structural and kinetic analysis of drug resistant mutants of HIV-1 protease
1KZK_A                                                                                                                                                                                   JE-2147-HIV Protease Complex
2HS1_A                                                                                                  Ultra-high resolution X-ray crystal structure of HIV-1 protease V32I mutant with TMC114 (darunavir) inhibitor
1K6C_A                                                                                                                                 LACK OF SYNERGY FOR INHIBITORS TARGETING A MULTI-DRUG RESISTANT HIV-1 PROTEASE
1MTB_A                                                                                                          Viability of a drug-resistant HIV-1 protease mutant: structural insights for better antiviral therapy
4Q1X_A                                                                   Mutations Outside the Active Site of HIV-1 Protease Alter Enzyme Structure and Dynamic Ensemble of the Active Site to Confer Drug Resistance
4Q1W_A                                                                   Mutations Outside the Active Site of HIV-1 Protease Alter Enzyme Structure and Dynamic Ensemble of the Active Site to Confer Drug Resistance
3D1X_A                                                                                                                                                Crystal structure of HIV-1 mutant I54M and inhibitor saquinavir
2AVM_A                                                         Kinetics, stability, and structural changes in high resolution crystal structures of HIV-1 protease with drug resistant mutations L24I, I50V, AND G73S
3PWM_A                                                                                                                                                                      HIV-1 Protease Mutant L76V with Darunavir
3KT2_A                                                                                                                                                                Crystal Structure of N88D mutant HIV-1 Protease
1SDV_A                                                                                                             Crystal structures of HIV protease V82A and L90M mutants reveal changes in indinavir binding site.
3JVW_A                                                                                                                                                     HIV-1 Protease Mutant G86A with symmetric inhibitor DMP323
3OY4_A                                                                                                                     Crystal Structure of HIV-1 L76V Protease in Complex with the Protease Inhibitor Darunavir.
1A94_A                                                                                                                                                       STRUCTURAL BASIS FOR SPECIFICITY OF RETROVIRAL PROTEASES
2HS2_A                                                                                                                           Crystal structure of M46L mutant of HIV-1 protease complexed with TMC114 (darunavir)
                                                             citation rObserved
1W5V_A                      Lindberg, J., et al. Eur J Biochem (2004)   0.18900
2FDE_A               Miller, J.F., et al. Bioorg Med Chem Lett (2006)   0.20400
1AJV_A                          Backbro, K., et al. J Med Chem (1997)   0.18700
2R38_A                      Bottcher, J., et al. J Mol Biology (2008)   0.18280
2R3T_A                      Bottcher, J., et al. J Mol Biology (2008)   0.17810
1HXB_A                            Krohn, A., et al. J Med Chem (1991)   0.18400
1BV9_A                          Ala, P.J., et al. Biochemistry (1998)   0.19800
1AAQ_A                       Dreyer, G.B., et al. Biochemistry (1992)   0.19000
1AXA_A                            Hong, L., et al. Protein Sci (1998)        NA
1HVS_A                   Baldwin, E.T., et al. Nat Struct Biol (1995)   0.15000
1ZP8_A                            Brik, A., et al. Chembiochem (2005)   0.19700
2QHC_A                       Saskova, K.G., et al. Protein Sci (2008)   0.19277
1A8G_A                     Ringhofer, S., et al. J Mol Biology (1999)   0.15000
2O4L_A                            Muzammil, S., et al. J Virol (2007)   0.19300
5COK_A                                Aoki, M., et al. J Virol (2015)   0.21220
1TCX_A                         Hoog, S.S., et al. Biochemistry (1996)   0.18000
2Z54_A                       Saskova, K.G., et al. Protein Sci (2008)   0.20215
1D4S_A                     Thaisrivongs, S., et al. J Med Chem (1996)        NA
1BV7_A                          Ala, P.J., et al. Biochemistry (1998)   0.19600
1BWA_A                          Ala, P.J., et al. Biochemistry (1998)   0.19900
1A9M_A                              Hong, L., et al. FEBS Lett (1997)   0.18500
2FLE_A                       Clemente, J.C., et al. J Med Chem (2008)   0.20700
1ODY_A                        Kervinen, J., et al. Protein Sci (1998)        NA
1GNN_A                           Hong, L., et al. Biochemistry (1996)        NA
1GNM_A                           Hong, L., et al. Biochemistry (1996)        NA
5YRS_B                           Das, A., et al. J Am Chem Soc (2010)   0.22300
5YRS_A                           Das, A., et al. J Am Chem Soc (2010)   0.22300
1HEF_E                  Murthy, K.H., et al. J Biological Chem (1992)   0.15900
1ODX_A                  Kervinen, J., et al. Protein Pept Lett (1996)        NA
4QGI_A                     Goldfarb, N.E., et al. Biochemistry (2015)   0.19010
1BVE_A                        Yamazaki, T., et al. Protein Sci (1996)        NA
2AZ8_A                       Heaslet, H., et al. J Mol Biology (2006)   0.23400
1A30_A                        Louis, J.M., et al. Biochemistry (1998)   0.18900
6DH6_A                   Lockbaum, G.J., et al. ACS Infect Dis (2019)   0.17740
6DH0_A                   Lockbaum, G.J., et al. ACS Infect Dis (2019)   0.18400
2I4D_A                        Cihlar, T., et al. J Mol Biology (2006)        NA
6OOS_A                          Henes, M., et al. Biochemistry (2019)   0.16520
1RL8_A                    Rezacova, P., et al. To be published (2005)   0.22600
1ZSF_A Duskova, J., et al. Acta Crystallogr D Biol Crystallogr (2006)   0.22180
2Q64_A                       Kozisek, M., et al. J Mol Biology (2007)   0.18280
6DH3_A                   Lockbaum, G.J., et al. ACS Infect Dis (2019)   0.17590
2NPH_A                Das, A., et al. Proc Natl Acad Sci U S A (2006)   0.21600
2Q63_A                       Kozisek, M., et al. J Mol Biology (2007)   0.19074
1LZQ_A                          Skalova, T., et al. J Med Chem (2003)   0.20800
1FB7_A                            Hong, L., et al. Protein Sci (2000)   0.18700
1G6L_A                             Pillai, B., et al. Proteins (2001)        NA
1HIV_A                          Thanki, N., et al. Protein Sci (1992)   0.16900
6OOU_A                          Henes, M., et al. Biochemistry (2019)   0.20640
1HVC_A                      Bhat, T.N., et al. Nat Struct Biol (1994)        NA
2I4V_A                        Cihlar, T., et al. J Mol Biology (2006)        NA
2AZ9_A                       Heaslet, H., et al. J Mol Biology (2006)   0.24800
6OOT_A                          Henes, M., et al. Biochemistry (2019)   0.19690
2P3B_B                       Sanches, M., et al. J Mol Biology (2007)   0.16500
5KAO_A           Amano, M., et al. Antimicrob Agents Chemother (2016)   0.18560
2WL0_A                     Mahalingam, A.K., et al. J Med Chem (2010)   0.21460
6OPT_A                         Henes, M., et al. ACS Chem Biol (2019)   0.18460
1IZI_A                         Weber, J., et al. J Mol Biology (2002)   0.19202
1MRX_A                               Vega, S., et al. Proteins (2004)        NA
2PYM_A                       Kozisek, M., et al. J Mol Biology (2007)   0.19523
2PYN_A                       Kozisek, M., et al. J Mol Biology (2007)   0.18993
1DMP_A                           Hodge, C.N., et al. Chem Biol (1996)   0.21000
4K4P_A              Tiefenbrunn, T., et al. Chem Biol Drug Des (2014)   0.18753
1LV1_A            Kumar, M., et al. Biochem Biophys Res Commun (2002)   0.19400
1AID_A                  Rutenber, E., et al. J Biological Chem (1993)   0.17400
1ZBG_A                           Duskova, J., et al. To be published    0.18498
3TKG_A                      Agniswamy, J., et al. Biochemistry (2012)   0.16290
5YOK_A                           Hidaka, K., et al. J Med Chem (2018)        NA
1FGC_C                         Mahalingam, B., et al. Proteins (2001)        NA
3K4V_A              Olajuyigbe, F.M., et al. ACS Med Chem Lett (2010)   0.15930
3KT5_A         Bihani, S.C., et al. Biochem Biophys Res Commun (2009)   0.18000
4QLH_A                                   Hosur, M.V. To be published    0.19530
2F3K_A Prabu-Jeyabalan, M., et al. Antimicrob Agents Chemother (2006)   0.16489
4Q5M_A                  Prashar, V., et al. Chem Biol Drug Des (2015)   0.20760
2AOC_A                                  Tie, Y., et al. FEBS J (2005)        NA
3B80_A                   Kovalevsky, A.Y., et al. Biochemistry (2007)   0.15300
3VF5_A                          Chang, Y.C., et al. J Med Chem (2012)   0.14680
2AVQ_A                           Liu, F., et al. J Mol Biology (2005)        NA
1DW6_C                    Mahalingam, B., et al. Eur J Biochem (1999)        NA
1KZK_A                      Reiling, K.K., et al. Biochemistry (2002)   0.15200
2HS1_A                  Kovalevsky, A.Y., et al. J Mol Biology (2006)   0.12420
1K6C_A                          King, N.M., et al. Protein Sci (2002)   0.19400
1MTB_A                     Prabu-Jeyabalan, M., et al. J Virol (2003)   0.19100
4Q1X_A                     Ragland, D.A., et al. J Am Chem Soc (2014)   0.18289
4Q1W_A                     Ragland, D.A., et al. J Am Chem Soc (2014)   0.17454
3D1X_A                           Liu, F., et al. J Mol Biology (2008)   0.15470
2AVM_A                           Liu, F., et al. J Mol Biology (2005)   0.10400
3PWM_A                        Louis, J.M., et al. Biochemistry (2011)   0.14000
3KT2_A         Bihani, S.C., et al. Biochem Biophys Res Commun (2009)   0.18700
1SDV_A                    Mahalingam, B., et al. Eur J Biochem (2004)   0.15970
3JVW_A                             Ishima, R., et al. Proteins (2009)   0.22560
3OY4_A                        Schiffer, C.A., et al. To be published    0.18026
1A94_A                             Wu, J., et al. Biochemistry (1998)   0.18200
2HS2_A                  Kovalevsky, A.Y., et al. J Mol Biology (2006)   0.13980
         rFree   rWork spaceGroup
1W5V_A 0.21500 0.18900  P 21 21 2
2FDE_A 0.28000 0.20400       P 61
1AJV_A 0.24700 0.18700  P 21 21 2
2R38_A 0.23790 0.18110  P 21 21 2
2R3T_A 0.23280 0.17680  P 21 21 2
1HXB_A      NA 0.18400       P 61
1BV9_A      NA 0.19800       P 61
1AAQ_A      NA      NA       P 61
1AXA_A      NA 0.19400       P 61
1HVS_A      NA 0.15000 P 21 21 21
1ZP8_A 0.32600 0.19700   P 61 2 2
2QHC_A 0.22774 0.19101       P 61
1A8G_A 0.24400 0.15000       P 61
2O4L_A 0.21600 0.19000  P 21 21 2
5COK_A 0.25520 0.21010       P 61
1TCX_A      NA 0.18000  P 21 21 2
2Z54_A 0.25407 0.19948       P 61
1D4S_A 0.29400      NA  P 21 21 2
1BV7_A      NA 0.19600       P 61
1BWA_A      NA 0.19900       P 61
1A9M_A      NA 0.18500    I 2 2 2
2FLE_A 0.25050 0.19900       P 61
1ODY_A      NA 0.16700       P 61
1GNN_A      NA 0.18300       P 61
1GNM_A      NA 0.17500       P 61
5YRS_B 0.25000 0.22300       P 61
5YRS_A 0.25000 0.22300       P 61
1HEF_E      NA      NA   P 61 2 2
1ODX_A      NA 0.16600       P 61
4QGI_A 0.24570 0.18710 P 21 21 21
1BVE_A      NA      NA       &lt;NA&gt;
2AZ8_A 0.28600 0.23400   P 61 2 2
1A30_A 0.22700 0.18900  P 21 21 2
6DH6_A 0.22190 0.17500 P 21 21 21
6DH0_A 0.21210 0.18250 P 21 21 21
2I4D_A 0.27400 0.24100  P 21 21 2
6OOS_A 0.19840 0.16350   P 1 21 1
1RL8_A 0.26900 0.22600       P 61
1ZSF_A 0.27580 0.22100       P 61
2Q64_A 0.25772 0.17924       P 61
6DH3_A 0.20960 0.17410 P 21 21 21
2NPH_A 0.25900 0.21400       P 61
2Q63_A 0.25005 0.18430       P 61
1LZQ_A 0.25200 0.20300       P 61
1FB7_A 0.24300 0.18700     I 21 3
1G6L_A 0.28400 0.19500       P 61
1HIV_A      NA      NA P 21 21 21
6OOU_A 0.25490 0.20380 P 21 21 21
1HVC_A      NA      NA P 21 21 21
2I4V_A 0.25600 0.21900  P 21 21 2
2AZ9_A 0.30200 0.24800   P 61 2 2
6OOT_A 0.23460 0.19170       P 61
2P3B_B 0.21900 0.15900       P 61
5KAO_A 0.22090 0.18380 P 21 21 21
2WL0_A 0.24400 0.21460  P 21 21 2
6OPT_A 0.22770 0.18240 P 21 21 21
1IZI_A 0.24797 0.18942       P 61
1MRX_A 0.25030 0.20770  P 21 21 2
2PYM_A 0.24378 0.19270       P 61
2PYN_A 0.23809 0.18743       P 61
1DMP_A      NA 0.21000       P 61
4K4P_A 0.23400 0.18513   I 41 2 2
1LV1_A 0.26000 0.19400       P 61
1AID_A      NA 0.17400       P 41
1ZBG_A 0.23732 0.18665    I 2 2 2
3TKG_A 0.21400 0.16000 P 21 21 21
5YOK_A 0.12500 0.10300  P 21 21 2
1FGC_C 0.26800 0.21200 P 21 21 21
3K4V_A 0.22650      NA   P 1 21 1
3KT5_A 0.22500 0.17800       P 61
4QLH_A 0.25597 0.19216       P 61
2F3K_A 0.19667 0.16312 P 21 21 21
4Q5M_A 0.23470 0.20610       P 61
2AOC_A 0.16590 0.12340  P 21 21 2
3B80_A 0.21870      NA  P 21 21 2
3VF5_A 0.16860      NA  P 21 21 2
2AVQ_A 0.14410 0.11140  P 21 21 2
1DW6_C 0.24900 0.21800 P 21 21 21
1KZK_A 0.18850 0.15060 P 21 21 21
2HS1_A 0.14900 0.12420 P 21 21 21
1K6C_A 0.23100 0.19400 P 21 21 21
1MTB_A 0.25700 0.19100 P 21 21 21
4Q1X_A 0.23410 0.18020 P 21 21 21
4Q1W_A 0.20634 0.17288 P 21 21 21
3D1X_A 0.14700 0.11900  P 21 21 2
2AVM_A 0.13200 0.10600  P 21 21 2
3PWM_A 0.18900      NA  P 21 21 2
3KT2_A 0.20800 0.18600       P 61
1SDV_A 0.20540 0.15970  P 21 21 2
3JVW_A 0.28880 0.22560  P 21 21 2
3OY4_A 0.21902 0.17814 P 21 21 21
1A94_A 0.28100 0.18200   P 1 1 21
2HS2_A 0.19470 0.13980 P 21 21 21</code></pre>
</div>
</div>
</section>
<section id="principal-component-analysis" class="level3">
<h3 class="anchored" data-anchor-id="principal-component-analysis">Principal component analysis</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>pc.xray <span class="ot">&lt;-</span> <span class="fu">pca</span>(pdbs)</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pc.xray)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate RMSD</span></span>
<span id="cb136-2"><a href="#cb136-2" aria-hidden="true" tabindex="-1"></a>rd <span class="ot">&lt;-</span> <span class="fu">rmsd</span>(pdbs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in rmsd(pdbs): No indices provided, using the 99 non NA positions</code></pre>
</div>
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Structure-based clustering</span></span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>hc.rd <span class="ot">&lt;-</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(rd))</span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>grps.rd <span class="ot">&lt;-</span> <span class="fu">cutree</span>(hc.rd, <span class="at">k=</span><span class="dv">3</span>)</span>
<span id="cb138-4"><a href="#cb138-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb138-5"><a href="#cb138-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(pc.xray, <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"grey50"</span>, <span class="at">bg=</span>grps.rd, <span class="at">pch=</span><span class="dv">21</span>, <span class="at">cex=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a>pc1 <span class="ot">&lt;-</span> <span class="fu">mktrj</span>(pc.xray, <span class="at">pc=</span><span class="dv">1</span>, <span class="at">file=</span><span class="st">"pc_1.pdb"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb140"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb140-1"><a href="#cb140-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb140-2"><a href="#cb140-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb140-3"><a href="#cb140-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-4"><a href="#cb140-4" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1=</span>pc.xray<span class="sc">$</span>z[,<span class="dv">1</span>], </span>
<span id="cb140-5"><a href="#cb140-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">PC2=</span>pc.xray<span class="sc">$</span>z[,<span class="dv">2</span>], </span>
<span id="cb140-6"><a href="#cb140-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">col=</span><span class="fu">as.factor</span>(grps.rd),</span>
<span id="cb140-7"><a href="#cb140-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">ids=</span>ids)</span>
<span id="cb140-8"><a href="#cb140-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb140-9"><a href="#cb140-9" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(df) <span class="sc">+</span> </span>
<span id="cb140-10"><a href="#cb140-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(PC1, PC2, <span class="at">col=</span>col, <span class="at">label=</span>ids) <span class="sc">+</span></span>
<span id="cb140-11"><a href="#cb140-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb140-12"><a href="#cb140-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">max.overlaps =</span> <span class="dv">20</span>) <span class="sc">+</span></span>
<span id="cb140-13"><a href="#cb140-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb140-14"><a href="#cb140-14" aria-hidden="true" tabindex="-1"></a>p</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: ggrepel: 65 unlabeled data points (too many overlaps). Consider
increasing max.overlaps</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a>modes <span class="ot">&lt;-</span> <span class="fu">nma</span>(pdbs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in nma.pdbs(pdbs): 1G6L_A.pdb, 1LV1_A.pdb, 3KT5_A.pdb, 4QLH_A.pdb, 4Q5M_A.pdb, 3KT2_A.pdb might have missing residue(s) in structure:
   Fluctuations at neighboring positions may be affected.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>
Details of Scheduled Calculation:
  ... 93 input structures 
  ... storing 291 eigenvectors for each structure 
  ... dimension of x$U.subspace: ( 297x291x93 )
  ... coordinate superposition prior to NM calculation 
  ... aligned eigenvectors (gap containing positions removed)  
  ... estimated memory usage of final 'eNMA' object: 61.7 Mb 


  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   1%
  |                                                                            
  |==                                                                    |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   4%
  |                                                                            
  |====                                                                  |   5%
  |                                                                            
  |=====                                                                 |   6%
  |                                                                            
  |=====                                                                 |   8%
  |                                                                            
  |======                                                                |   9%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |===========                                                           |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |============                                                          |  17%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  19%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  22%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |=================                                                     |  24%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |===================                                                   |  27%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |=======================                                               |  32%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |==========================                                            |  37%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |===========================                                           |  39%
  |                                                                            
  |============================                                          |  40%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |=============================                                         |  42%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  45%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |==================================                                    |  48%
  |                                                                            
  |===================================                                   |  49%
  |                                                                            
  |===================================                                   |  51%
  |                                                                            
  |====================================                                  |  52%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |======================================                                |  55%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |=========================================                             |  58%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  60%
  |                                                                            
  |===========================================                           |  61%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |============================================                          |  63%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |===============================================                       |  68%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |===================================================                   |  73%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |=====================================================                 |  76%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |=======================================================               |  78%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |========================================================              |  81%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |==========================================================            |  83%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |===========================================================           |  85%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  91%
  |                                                                            
  |=================================================================     |  92%
  |                                                                            
  |=================================================================     |  94%
  |                                                                            
  |==================================================================    |  95%
  |                                                                            
  |===================================================================   |  96%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |====================================================================  |  98%
  |                                                                            
  |===================================================================== |  99%
  |                                                                            
  |======================================================================| 100%</code></pre>
</div>
<div class="sourceCode cell-code" id="cb145"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb145-1"><a href="#cb145-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(modes, pdbs, <span class="at">col=</span>grps.rd)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Extracting SSE from pdbs$sse attribute</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="prediction-of-functional-motions" class="level2">
<h2 class="anchored" data-anchor-id="prediction-of-functional-motions">Prediction of functional motions</h2>
<p>We can run an Normal Mode Analysis (NMA) to predict large scale motions/flexibility/dynamics of any biomolecule that we can read into R.</p>
<p>Let’s look at ADK</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb147"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb147-1"><a href="#cb147-1" aria-hidden="true" tabindex="-1"></a>adk <span class="ot">&lt;-</span> <span class="fu">read.pdb</span>(<span class="st">"1ake"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Note: Accessing on-line PDB file
   PDB has ALT records, taking A only, rm.alt=TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb149"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb149-1"><a href="#cb149-1" aria-hidden="true" tabindex="-1"></a>adk_A <span class="ot">&lt;-</span> <span class="fu">trim.pdb</span>(adk, <span class="at">chain =</span> <span class="st">"A"</span>)</span>
<span id="cb149-2"><a href="#cb149-2" aria-hidden="true" tabindex="-1"></a>adk_A</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
 Call:  trim.pdb(pdb = adk, chain = "A")

   Total Models#: 1
     Total Atoms#: 1954,  XYZs#: 5862  Chains#: 1  (values: A)

     Protein Atoms#: 1656  (residues/Calpha atoms#: 214)
     Nucleic acid Atoms#: 0  (residues/phosphate atoms#: 0)

     Non-protein/nucleic Atoms#: 298  (residues: 242)
     Non-protein/nucleic resid values: [ AP5 (1), HOH (241) ]

   Protein sequence:
      MRIILLGAPGAGKGTQAQFIMEKYGIPQISTGDMLRAAVKSGSELGKQAKDIMDAGKLVT
      DELVIALVKERIAQEDCRNGFLLDGFPRTIPQADAMKEAGINVDYVLEFDVPDELIVDRI
      VGRRVHAPSGRVYHVKFNPPKVEGKDDVTGEELTTRKDDQEETVRKRLVEYHQMTAPLIG
      YYSKEAEAGNTKYAKVDGTKPVAEVRADLEKILG

+ attr: atom, helix, sheet, seqres, xyz,
        calpha, call</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb151"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb151-1"><a href="#cb151-1" aria-hidden="true" tabindex="-1"></a>m <span class="ot">&lt;-</span> <span class="fu">nma</span>(adk_A)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Building Hessian...        Done in 0.03 seconds.
 Diagonalizing Hessian...   Done in 0.47 seconds.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb153"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb153-1"><a href="#cb153-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(m)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class09_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Let’s write a “trajectory” of predicted motion</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb154"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb154-1"><a href="#cb154-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mktrj</span>(m,<span class="at">file=</span><span class="st">"adk_nma.pdb"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="play-with-3d-viewing-in-r" class="level2">
<h2 class="anchored" data-anchor-id="play-with-3d-viewing-in-r">Play with 3D viewing in R</h2>
<p>We can use the new <strong>bio3dview</strong> package, which is not yet on CRAN, to render interactive 3D views in R and HTML quarto output reports.</p>
<p>To install from GitHub, we can use the <strong>pak</strong> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb155"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb155-1"><a href="#cb155-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pak)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'pak' was built under R version 4.5.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb157"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb157-1"><a href="#cb157-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bio3d)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>