---
title: "Class 13"
author: "Kavi (PID: A69046927)"
format: pdf
toc: true
---

## Background

The data for for hands-on session comes from GEO entry: GSE37704, which is associated with the following publication:

Trapnell C, Hendrickson DG, Sauvageau M, Goff L et al. "Differential analysis of gene regulation at transcript resolution with RNA-seq". Nat Biotechnol 2013 Jan;31(1):46-53. PMID: 23222703
The authors report on differential analysis of lung fibroblasts in response to loss of the developmental transcription factor HOXA1. Their results and others indicate that HOXA1 is required for lung fibroblast and HeLa cell cycle progression. In particular their analysis show that "loss of HOXA1 results in significant expression level changes in thousands of individual transcripts, along with isoform switching events in key regulators of the cell cycle". For our session we have used their Sailfish gene-level estimated counts and hence are restricted to protein-coding genes only.

## Data Import

```{r,message=F}
library(DESeq2)
```
```{r}
metaFile <- "GSE37704_metadata.csv"
countFile <- "GSE37704_featurecounts.csv"
metadata = read.csv(metaFile, row.names=1)
head(metadata)
counts = read.csv(countFile, row.names=1)
head(counts)
```
Check correspondence of `metadata` and `counts` (i.e. that the columns in `counts` match the rows in the `metadata`)
```{r}
metadata
```
```{r}
head(counts)
```

> Q1. Complete the code below to remove the troublesome first column from countData

Fix to remove that first "length" colmn of `counts`.
```{r}
counts <- counts[,-1]
```
```{r}
tot.counts <- rowSums(counts)
head(tot.counts)
```
Let;s remove all zero count genes

```{r}
zero.inds <- tot.counts == 0
head(zero.inds)
```
> Q2. Complete the code below to filter countData to exclude genes (i.e. rows) where we have 0 read count across all samples (i.e. columns).

```{r}
head(counts[!zero.inds,])
```

```{r}
colnames(counts)
test_cols <- all(colnames(counts)[-1] == metadata$id)
```
```{r}
if(test_cols) {
  message("Wow... there is a problem with the metadata counts setup")
}
```

## Setup for DESeq
```{r}
library(DESeq2)
```
```{r}
dds <- DESeqDataSetFromMatrix(countData = counts,
                              colData = metadata,
                              design = ~condition)
```

## Run DESeq

```{r}
dds <- DESeq(dds)
```

## Get results

> Q3. Call the summary() function on your results to get a sense of how many genes are up or down-regulated at the default 0.1 p-value cutoff.

```{r}
res <- results(dds)
summary(res)
head(res)
```

## Add annotation
> Q. Use the mapIDs() function multiple times to add SYMBOL, ENTREZID and GENENAME annotation to our results by completing the code below.

```{r}
library("AnnotationDbi")
library("org.Hs.eg.db")
res$symbol <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="SYMBOL",          # The new format we want to add
                     multiVals="first")
res$genename <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="GENENAME",          # The new format we want to add
                     multiVals="first")
res$entrezid <- mapIds(org.Hs.eg.db,
                     keys=row.names(res), # Our genenames
                     keytype="ENSEMBL",        # The format of our genenames
                     column="ENTREZID",          # The new format we want to add
                     multiVals="first")
head(res)
```


## Visualize results

> Q4. Improve this plot by completing the below code, which adds color and axis labels

```{r}
my_cols <- rep("gray", nrow(res))
my_cols[which(res$padj < 0.05 & res$log2FoldChange > 1)] <- "red"
my_cols[which(res$padj < 0.05 & res$log2FoldChange < -1)] <- "blue"
```

```{r}
library(ggplot2)
library(ggrepel)

ggplot(res, aes(x = log2FoldChange, y = -log10(pvalue))) +
  geom_point(col = my_cols, alpha = 0.6) +
  theme_classic() +
  geom_vline(xintercept = c(-1, 1), linetype = "dashed", color = "red") +
  geom_hline(yintercept = -log10(0.05), linetype = "dashed", color = "red") + 
  labs(x = "Log2 Fold Change", y = "-Log10 P-value")
  # Only label significant hits:
  geom_text_repel(
    data = subset(res, padj < 0.05 & abs(log2FoldChange) > 1),
    aes(label = genename),
    size = 3
  )
```

## Pathway analysis
```{r}
library(gage)
library(gageData)
foldchanges <- res$log2FoldChange
names(foldchanges) <- res$entrezid
head(foldchanges)
data(kegg.sets.hs)
keggres = gage(foldchanges, gsets=kegg.sets.hs)
```
> Q5. Can you do the same procedure as above to plot the pathview figures for the top 5 down-reguled pathways?

```{r}
head(keggres$less,5)
```
```{r}
library(pathview)
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```
![Cell cycle](hsa04110.pathview.png)
### GO Analysis

Let's try GO analysis and compare to KEGG analysis

```{r}
data(go.sets.hs)
data(go.subs.hs)

gobpsets = go.sets.hs[go.subs.hs$BP]
gobpres = gage(foldchanges, gsets=gobpsets)
head(gobpres$less, 5)
```
### Reactome

Some folks really like Reactome online (i.e. their webpage viewer) rather than the R package of the same name (available from Bioconductor).

To use the website viewer we want to upload our set of gene symbols for the genes we want to focus on.

```{r}
sig_genes <- res$symbol[which(res$padj < 0.05 & !is.na(res$padj))]

write.table(sig_genes, file="significant_genes.txt",
            row.names=FALSE, col.names=FALSE, quote=FALSE)
```

> Q6. What pathway has the most significant “Entities p-value”? Do the most significant pathways listed match your previous KEGG results? What factors could cause differences between the two methods?

"Organelle fission" has the most significant value, where it was cell cycle before. Differences between KEGG and Reactome pathway analysis results can arise from variations in pathway definitions, gene set composition, statistical methods, and biological context considered by each database.


## Save results
```{r}
write.csv(as.data.frame(res), file="DESeq2_results_annotated.csv")
save(res, counts, metadata, file="DESeq2_results_annotated.RData")
```
